--- origsrc/pidgin-2.10.11/configure.ac	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/configure.ac	2015-03-19 12:40:17.646294400 -0500
@@ -2515,6 +2515,33 @@ if test "x$enable_debug" = "xyes" ; then
 	AC_DEFINE(DEBUG, 1, [Define if debugging is enabled.])
 fi
 
+# win32
+case "$host_os" in
+cygwin*)
+	platform_win32=yes
+	LIBFINCH="-lfinch"
+	LIBPIDGIN="-lpidgin"
+	FINCH_PLUGIN_LDFLAGS="-no-undefined -Wl,\$(top_builddir)/finch/libfinch.dll.a \$(top_builddir)/finch/libgnt/libgnt.la \$(top_builddir)/libpurple/libpurple.la"
+	PIDGIN_PLUGIN_LDFLAGS="-no-undefined -Wl,\$(top_builddir)/pidgin/libpidgin.dll.a \$(top_builddir)/libpurple/libpurple.la"
+	PURPLE_PLUGIN_LDFLAGS="-no-undefined \$(top_builddir)/libpurple/libpurple.la"
+	;;
+*)
+	platform_win32=no
+	LIBFINCH=
+	LIBPIDGIN=
+	FINCH_PLUGIN_LDFLAGS=
+	PIDGIN_PLUGIN_LDFLAGS=
+	PURPLE_PLUGIN_LDFLAGS=
+	;;
+esac
+
+AM_CONDITIONAL(PLATFORM_WIN32, test "x$platform_win32" = "xyes")
+AC_SUBST(LIBFINCH)
+AC_SUBST(LIBPIDGIN)
+AC_SUBST(FINCH_PLUGIN_LDFLAGS)
+AC_SUBST(PIDGIN_PLUGIN_LDFLAGS)
+AC_SUBST(PURPLE_PLUGIN_LDFLAGS)
+
 AM_CONDITIONAL(PURPLE_AVAILABLE, true)
 
 AC_CONFIG_FILES([Makefile
--- origsrc/pidgin-2.10.11/finch/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/finch/Makefile.am	2015-03-19 12:40:17.662296500 -0500
@@ -7,7 +7,7 @@ EXTRA_DIST = \
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = finch.pc
 
-SUBDIRS = libgnt plugins
+SUBDIRS = libgnt . plugins
 
 if ENABLE_GNT
 
@@ -64,7 +64,13 @@ finchinclude_HEADERS = \
 	$(finch_headers)
 
 finch_DEPENDENCIES = @LIBOBJS@
+
+if PLATFORM_WIN32
+finch_LDFLAGS = -Wl,--export-all-symbols -Wl,--out-implib,libfinch.dll.a
+else
 finch_LDFLAGS = -export-dynamic
+endif
+
 finch_LDADD = \
 	@LIBOBJS@ \
 	$(DBUS_LIBS) \
@@ -91,3 +97,9 @@ AM_CPPFLAGS = \
 	$(LIBXML_CFLAGS) \
 	$(GSTREAMER_CFLAGS) \
 	$(GNT_CFLAGS)
+
+if PLATFORM_WIN32
+install-data-local:
+	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) libfinch.dll.a $(DESTDIR)$(libdir)
+endif
--- origsrc/pidgin-2.10.11/finch/finch.pc.in	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/finch/finch.pc.in	2015-03-19 12:40:17.652295200 -0500
@@ -11,4 +11,5 @@ Description: Finch is an instant messeng
 Version: @VERSION@
 Requires: gnt purple
 Cflags: -I${includedir}/finch
+Libs: -L${libdir} @LIBFINCH@
 
--- origsrc/pidgin-2.10.11/finch/libgnt/wms/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/finch/libgnt/wms/Makefile.am	2015-03-19 12:40:17.657295800 -0500
@@ -6,8 +6,8 @@ else
 purple_wms =
 endif
 
-s_la_LDFLAGS             = -module -avoid-version
-irssi_la_LDFLAGS         = -module -avoid-version
+s_la_LDFLAGS             = -module -avoid-version -no-undefined
+irssi_la_LDFLAGS         = -module -avoid-version -no-undefined
 
 plugin_LTLIBRARIES = \
 	$(wms) \
--- origsrc/pidgin-2.10.11/finch/plugins/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/finch/plugins/Makefile.am	2015-03-19 12:40:17.666797000 -0500
@@ -1,9 +1,9 @@
-gntclipboard_la_LDFLAGS = -module -avoid-version
-gntgf_la_LDFLAGS      = -module -avoid-version
-gnthistory_la_LDFLAGS = -module -avoid-version
-gntlastlog_la_LDFLAGS = -module -avoid-version
-gnttinyurl_la_LDFLAGS = -module -avoid-version
-grouping_la_LDFLAGS   = -module -avoid-version
+gntclipboard_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gntgf_la_LDFLAGS      = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gnthistory_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gntlastlog_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gnttinyurl_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+grouping_la_LDFLAGS   = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.10.11/libpurple/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/Makefile.am	2015-03-19 12:40:17.669797400 -0500
@@ -29,7 +29,7 @@ if USE_GCONFTOOL
 GCONF_DIR=data/gconf
 endif
 
-SUBDIRS = $(GCONF_DIR) plugins protocols ciphers . tests example
+SUBDIRS = ciphers . $(GCONF_DIR) plugins protocols tests example
 
 purple_coresources = \
 	account.c \
--- origsrc/pidgin-2.10.11/libpurple/plugins/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/plugins/Makefile.am	2015-03-19 12:40:17.674298000 -0500
@@ -24,27 +24,27 @@ SUBDIRS = \
 
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-autoaccept_la_LDFLAGS       = -module -avoid-version
-buddynote_la_LDFLAGS        = -module -avoid-version
-ciphertest_la_LDFLAGS		= -module -avoid-version
-codeinline_la_LDFLAGS		= -module -avoid-version
-debug_example_la_LDFLAGS    = -module -avoid-version
-helloworld_la_LDFLAGS       = -module -avoid-version
-idle_la_LDFLAGS             = -module -avoid-version
-joinpart_la_LDFLAGS         = -module -avoid-version
-log_reader_la_LDFLAGS       = -module -avoid-version
-newline_la_LDFLAGS          = -module -avoid-version
-notify_example_la_LDFLAGS   = -module -avoid-version
-offlinemsg_la_LDFLAGS       = -module -avoid-version
-one_time_password_la_LDFLAGS	= -module -avoid-version
-pluginpref_example_la_LDFLAGS = -module -avoid-version
-psychic_la_LDFLAGS          = -module -avoid-version
-signals_test_la_LDFLAGS		= -module -avoid-version
-simple_la_LDFLAGS			= -module -avoid-version
-statenotify_la_LDFLAGS      = -module -avoid-version
+autoaccept_la_LDFLAGS       = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+buddynote_la_LDFLAGS        = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+ciphertest_la_LDFLAGS		= -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+codeinline_la_LDFLAGS		= -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+debug_example_la_LDFLAGS    = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+helloworld_la_LDFLAGS       = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+idle_la_LDFLAGS             = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+joinpart_la_LDFLAGS         = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+log_reader_la_LDFLAGS       = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+newline_la_LDFLAGS          = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+notify_example_la_LDFLAGS   = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+offlinemsg_la_LDFLAGS       = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+one_time_password_la_LDFLAGS	= -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+pluginpref_example_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+psychic_la_LDFLAGS          = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+signals_test_la_LDFLAGS		= -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+simple_la_LDFLAGS			= -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+statenotify_la_LDFLAGS      = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 # this can't be in a conditional otherwise automake 1.4 yells
-dbus_example_la_LDFLAGS     = -module -avoid-version
+dbus_example_la_LDFLAGS     = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.10.11/libpurple/plugins/mono/loader/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/plugins/mono/loader/Makefile.am	2015-03-19 12:40:17.677298400 -0500
@@ -12,7 +12,7 @@ mono_la_SOURCES	= \
 	blist-glue.c \
 	status-glue.c 
 
-mono_la_LDFLAGS  = -module -avoid-version
+mono_la_LDFLAGS  = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 mono_la_LIBADD = $(MONO_LIBS)
 
--- origsrc/pidgin-2.10.11/libpurple/plugins/perl/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/plugins/perl/Makefile.am	2015-03-19 12:40:17.685299400 -0500
@@ -4,7 +4,7 @@ perl_dirs = common
 
 plugin_LTLIBRARIES = perl.la
 
-perl_la_LDFLAGS = -module -avoid-version
+perl_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 perl_la_LIBADD = $(GLIB_LIBS) $(PERL_LIBS)
 perl_la_SOURCES = \
 	perl.c \
--- origsrc/pidgin-2.10.11/libpurple/plugins/perl/common/Makefile.PL.in	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/plugins/perl/common/Makefile.PL.in	2015-03-19 12:40:17.680798800 -0500
@@ -12,6 +12,7 @@ WriteMakefile(
     'DEFINE'            => '@DEBUG_CFLAGS@',
     'dynamic_lib'       => { 'OTHERLDFLAGS' => '@LDFLAGS@' },
     'INC'               => '-I. -I@srcdir@ -I@top_srcdir@ -I@top_srcdir@/libpurple @GLIB_CFLAGS@',
+    'LIBS'              => [ '-L@top_builddir@/libpurple/.libs -lpurple @GLIB_LIBS@' ],
     'OBJECT'            => '$(O_FILES)', # link all the C files too
 #    'OPTIMIZE'          => '-g', # For debugging
     'INSTALLDIRS'          => 'vendor',
--- origsrc/pidgin-2.10.11/libpurple/plugins/ssl/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/plugins/ssl/Makefile.am	2015-03-19 13:02:31.547678400 -0500
@@ -3,10 +3,10 @@ EXTRA_DIST = \
 
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-ssl_la_LDFLAGS        = -module -avoid-version
-ssl_gnutls_la_LDFLAGS = -module -avoid-version
-ssl_nss_la_LDFLAGS    = -module -avoid-version
-nss_prefs_la_LDFLAGS  = -module -avoid-version
+ssl_la_LDFLAGS        = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+ssl_gnutls_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+ssl_nss_la_LDFLAGS    = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+nss_prefs_la_LDFLAGS  = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.10.11/libpurple/plugins/tcl/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/plugins/tcl/Makefile.am	2015-03-19 12:40:17.694800600 -0500
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-tcl_la_LDFLAGS = -module -avoid-version
+tcl_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 plugin_LTLIBRARIES = tcl.la
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/bonjour/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/bonjour/Makefile.am	2015-03-19 12:40:17.697801000 -0500
@@ -25,7 +25,7 @@ BONJOURSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libbonjour_la_LDFLAGS = -module -avoid-version
+libbonjour_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_BONJOUR
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/gg/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/gg/Makefile.am	2015-03-19 12:40:17.700801400 -0500
@@ -113,7 +113,7 @@ GGSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libgg_la_LDFLAGS = -module -avoid-version
+libgg_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_GG
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/irc/Makefile.am	2014-11-23 10:41:25.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/irc/Makefile.am	2015-03-19 12:40:17.705301900 -0500
@@ -13,7 +13,7 @@ IRCSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libirc_la_LDFLAGS = -module -avoid-version
+libirc_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_IRC
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/jabber/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/jabber/Makefile.am	2015-03-19 12:40:17.708302300 -0500
@@ -95,7 +95,7 @@ JABBERSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libxmpp_la_LDFLAGS = -module -avoid-version
+libxmpp_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if USE_CYRUS_SASL
 JABBERSOURCES += auth_cyrus.c
@@ -111,7 +111,8 @@ libjabber_la_CFLAGS  = $(AM_CFLAGS)
 else
 
 st =
-pkg_LTLIBRARIES      = libjabber.la libxmpp.la
+noinst_LTLIBRARIES   = libjabber.la
+pkg_LTLIBRARIES      = libxmpp.la
 libjabber_la_SOURCES = $(JABBERSOURCES)
 libjabber_la_LIBADD  = $(GLIB_LIBS) $(SASL_LIBS) $(LIBXML_LIBS) $(IDN_LIBS)\
 	$(FARSIGHT_LIBS) \
--- origsrc/pidgin-2.10.11/libpurple/protocols/msn/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/msn/Makefile.am	2015-03-19 12:40:17.712802900 -0500
@@ -77,7 +77,7 @@ MSNSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libmsn_la_LDFLAGS = -module -avoid-version
+libmsn_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_MSN
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/mxit/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/mxit/Makefile.am	2015-03-19 12:40:17.717303500 -0500
@@ -40,7 +40,7 @@ MXITSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libmxit_la_LDFLAGS = -module -avoid-version
+libmxit_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_MXIT
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/myspace/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/myspace/Makefile.am	2015-03-19 12:40:17.721804000 -0500
@@ -19,7 +19,7 @@ MSIMSOURCES = markup.c \
 
 AM_CFLAGS = $(st)
 
-libmyspace_la_LDFLAGS = -module -avoid-version
+libmyspace_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_MYSPACE
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/novell/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/novell/Makefile.am	2015-03-19 12:40:17.724804400 -0500
@@ -28,7 +28,7 @@ NOVELLSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libnovell_la_LDFLAGS = -module -avoid-version
+libnovell_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_NOVELL
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/null/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/null/Makefile.am	2015-03-19 12:40:17.727804800 -0500
@@ -8,7 +8,7 @@ NULLSOURCES = nullprpl.c
 
 AM_CFLAGS = $(st)
 
-libnull_la_LDFLAGS = -module -avoid-version
+libnull_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 # nullprpl isn't built by default. when it is built, it's dynamically linked.
 st =
--- origsrc/pidgin-2.10.11/libpurple/protocols/oscar/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/oscar/Makefile.am	2015-03-19 12:40:17.730805200 -0500
@@ -50,8 +50,8 @@ OSCARSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libaim_la_LDFLAGS = -module -avoid-version
-libicq_la_LDFLAGS = -module -avoid-version
+libaim_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+libicq_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 if STATIC_OSCAR
 
 st = -DPURPLE_STATIC_PRPL
@@ -62,7 +62,8 @@ liboscar_la_CFLAGS  = $(AM_CFLAGS)
 else
 
 st =
-pkg_LTLIBRARIES     = liboscar.la libaim.la libicq.la
+noinst_LTLIBRARIES  = liboscar.la
+pkg_LTLIBRARIES     = libaim.la libicq.la
 liboscar_la_SOURCES = $(OSCARSOURCES)
 liboscar_la_LIBADD  = $(GLIB_LIBS)
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/sametime/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/sametime/Makefile.am	2015-03-19 12:40:17.735305700 -0500
@@ -24,7 +24,7 @@ pkg_LTLIBRARIES     = libsametime.la
 endif
 
 libsametime_la_SOURCES = $(SAMETIMESOURCES)
-libsametime_la_LDFLAGS = -module -avoid-version
+libsametime_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 libsametime_la_LIBADD = $(GLIB_LIBS) $(MEANWHILE_LIBS)
 
 AM_CPPFLAGS = \
--- origsrc/pidgin-2.10.11/libpurple/protocols/silc/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/silc/Makefile.am	2015-03-19 12:40:17.738306100 -0500
@@ -19,7 +19,7 @@ SILCSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libsilcpurple_la_LDFLAGS = -module -avoid-version
+libsilcpurple_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_SILC
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/silc10/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/silc10/Makefile.am	2015-03-19 12:40:17.741306500 -0500
@@ -9,7 +9,7 @@ SILCSOURCES = silc.c silcpurple.h buddy.
 
 AM_CFLAGS = $(st)
 
-libsilcpurple_la_LDFLAGS = -module -avoid-version
+libsilcpurple_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_SILC
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/simple/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/simple/Makefile.am	2015-03-19 12:40:17.744306900 -0500
@@ -11,7 +11,7 @@ SIMPLESOURCES = \
 
 AM_CFLAGS = $(st)
 
-libsimple_la_LDFLAGS = -module -avoid-version
+libsimple_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_SIMPLE
 
--- origsrc/pidgin-2.10.11/libpurple/protocols/yahoo/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/yahoo/Makefile.am	2015-03-19 12:40:17.747307300 -0500
@@ -27,8 +27,8 @@ YAHOOSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libyahoo_la_LDFLAGS = -module -avoid-version
-libyahoojp_la_LDFLAGS = -module -avoid-version
+libyahoo_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+libyahoojp_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_YAHOO
 
@@ -40,7 +40,8 @@ libymsg_la_CFLAGS  = $(AM_CFLAGS)
 else
 
 st =
-pkg_LTLIBRARIES     = libymsg.la libyahoo.la libyahoojp.la
+noinst_LTLIBRARIES  = libymsg.la
+pkg_LTLIBRARIES     = libyahoo.la libyahoojp.la
 
 libymsg_la_SOURCES = $(YAHOOSOURCES)
 libymsg_la_LIBADD  = $(GLIB_LIBS)
--- origsrc/pidgin-2.10.11/libpurple/protocols/zephyr/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/libpurple/protocols/zephyr/Makefile.am	2015-03-19 12:40:17.750307600 -0500
@@ -72,7 +72,7 @@ ZEPHYRLIBS = $(KRB4_LDFLAGS) $(KRB4_LIBS
 
 ZEPHYRLIBSEXT = $(ZEPHYR_LDFLAGS) $(ZEPHYR_LIBS)
 
-libzephyr_la_LDFLAGS = -module -avoid-version
+libzephyr_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_ZEPHYR
 
--- origsrc/pidgin-2.10.11/pidgin/Makefile.am	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/pidgin/Makefile.am	2015-03-19 12:40:17.753808100 -0500
@@ -33,7 +33,7 @@ EXTRA_DIST = \
 
 if ENABLE_GTK
 
-SUBDIRS = pixmaps plugins
+SUBDIRS = . pixmaps plugins
 
 bin_PROGRAMS = pidgin
 
@@ -145,7 +145,13 @@ pidgininclude_HEADERS = \
 
 
 pidgin_DEPENDENCIES = @LIBOBJS@
+
+if PLATFORM_WIN32
+pidgin_LDFLAGS = -Wl,--export-all-symbols,--out-implib,libpidgin.dll.a
+else
 pidgin_LDFLAGS = -export-dynamic
+endif
+
 pidgin_LDADD = \
 	@LIBOBJS@ \
 	$(GLIB_LIBS) \
@@ -200,3 +206,9 @@ CLEANFILES = $(appdata_DATA)
 DISTCLEANFILES = $(DESKTOP_FILE)
 
 endif  # ENABLE_GTK
+
+if PLATFORM_WIN32
+install-data-local:
+	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) libpidgin.dll.a $(DESTDIR)$(libdir)
+endif
--- origsrc/pidgin-2.10.11/pidgin/data/pidgin.pc.in	2014-11-23 10:41:26.000000000 -0600
+++ src/pidgin-2.10.11/pidgin/data/pidgin.pc.in	2015-03-19 12:40:17.757808600 -0500
@@ -13,4 +13,5 @@ Description: Pidgin is a GTK2-based inst
 Version: @VERSION@
 Requires: gtk+-2.0 purple
 Cflags: -I${includedir}/pidgin
+Libs: -L${libdir} @LIBPIDGIN@
 
--- origsrc/pidgin-2.10.11/pidgin/plugins/Makefile.am	2014-11-23 10:41:27.000000000 -0600
+++ src/pidgin-2.10.11/pidgin/plugins/Makefile.am	2015-03-19 12:40:17.773310600 -0500
@@ -31,25 +31,25 @@ SUBDIRS = \
 
 plugindir = $(libdir)/pidgin
 
-convcolors_la_LDFLAGS       = -module -avoid-version
-contact_priority_la_LDFLAGS = -module -avoid-version
-extplacement_la_LDFLAGS     = -module -avoid-version
-gtk_signals_test_la_LDFLAGS = -module -avoid-version
-gtkbuddynote_la_LDFLAGS     = -module -avoid-version
-history_la_LDFLAGS          = -module -avoid-version
-iconaway_la_LDFLAGS         = -module -avoid-version
-markerline_la_LDFLAGS       = -module -avoid-version
-notify_la_LDFLAGS           = -module -avoid-version
-pidginrc_la_LDFLAGS         = -module -avoid-version
-relnot_la_LDFLAGS           = -module -avoid-version
-sendbutton_la_LDFLAGS       = -module -avoid-version
-spellchk_la_LDFLAGS         = -module -avoid-version
-themeedit_la_LDFLAGS        = -module -avoid-version
-timestamp_la_LDFLAGS        = -module -avoid-version
-timestamp_format_la_LDFLAGS = -module -avoid-version
-unity_la_LDFLAGS            = -module -avoid-version
-vvconfig_la_LDFLAGS         = -module -avoid-version
-xmppconsole_la_LDFLAGS      = -module -avoid-version
+convcolors_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+contact_priority_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+extplacement_la_LDFLAGS     = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+gtk_signals_test_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+gtkbuddynote_la_LDFLAGS     = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+history_la_LDFLAGS          = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+iconaway_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+markerline_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+notify_la_LDFLAGS           = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+pidginrc_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+relnot_la_LDFLAGS           = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+sendbutton_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+spellchk_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+themeedit_la_LDFLAGS        = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+timestamp_la_LDFLAGS        = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+timestamp_format_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+unity_la_LDFLAGS            = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+vvconfig_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+xmppconsole_la_LDFLAGS      = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.10.11/pidgin/plugins/cap/Makefile.am	2014-11-23 10:41:27.000000000 -0600
+++ src/pidgin-2.10.11/pidgin/plugins/cap/Makefile.am	2015-03-19 12:40:17.760809000 -0500
@@ -1,6 +1,6 @@
 capdir = $(libdir)/pidgin
 
-cap_la_LDFLAGS = -module -avoid-version
+cap_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.10.11/pidgin/plugins/disco/Makefile.am	2014-11-23 10:41:27.000000000 -0600
+++ src/pidgin-2.10.11/pidgin/plugins/disco/Makefile.am	2015-03-19 12:40:17.764309400 -0500
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/pidgin
 
-xmppdisco_la_LDFLAGS = -module -avoid-version
+xmppdisco_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 EXTRA_DIST = \
         Makefile.mingw
--- origsrc/pidgin-2.10.11/pidgin/plugins/gestures/Makefile.am	2014-11-23 10:41:27.000000000 -0600
+++ src/pidgin-2.10.11/pidgin/plugins/gestures/Makefile.am	2015-03-19 12:40:17.767309800 -0500
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/pidgin
 
-gestures_la_LDFLAGS = -module -avoid-version
+gestures_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.10.11/pidgin/plugins/gevolution/Makefile.am	2014-11-23 10:41:27.000000000 -0600
+++ src/pidgin-2.10.11/pidgin/plugins/gevolution/Makefile.am	2015-03-19 12:40:17.770310200 -0500
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/pidgin
 
-gevolution_la_LDFLAGS = -module -avoid-version
+gevolution_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.10.11/pidgin/plugins/musicmessaging/Makefile.am	2014-11-23 10:41:27.000000000 -0600
+++ src/pidgin-2.10.11/pidgin/plugins/musicmessaging/Makefile.am	2015-03-19 12:40:17.776811000 -0500
@@ -3,7 +3,7 @@ EXTRA_DIST = \
 
 musicmessagingdir = $(libdir)/pidgin
 
-musicmessaging_la_LDFLAGS = -module -avoid-version
+musicmessaging_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 if ENABLE_DBUS
--- origsrc/pidgin-2.10.11/pidgin/plugins/perl/common/Makefile.PL.in	2014-11-23 10:41:27.000000000 -0600
+++ src/pidgin-2.10.11/pidgin/plugins/perl/common/Makefile.PL.in	2015-03-19 12:40:17.779811400 -0500
@@ -11,6 +11,7 @@ WriteMakefile(
     'DEFINE'        => '@DEBUG_CFLAGS@',
     'dynamic_lib'       => { 'OTHERLDFLAGS' => '@LDFLAGS@' },
     'INC'           => '-I. -I@srcdir@ -I@top_srcdir@ -I@top_srcdir@/libpurple -I@top_srcdir@/pidgin @GTK_CFLAGS@',
+    'LIBS'          => ['-L@top_builddir@/libpurple/.libs -lpurple -L@top_builddir@/pidgin @LIBPIDGIN@ @GLIB_LIBS@'],
     'OBJECT'        => '$(O_FILES)', # link all the C files too
     'TYPEMAPS'      => ["@top_srcdir@/libpurple/plugins/perl/common/typemap"],
 #    'OPTIMIZE'      => '-g', # For debugging.
--- origsrc/pidgin-2.10.11/pidgin/plugins/ticker/Makefile.am	2014-11-23 10:41:27.000000000 -0600
+++ src/pidgin-2.10.11/pidgin/plugins/ticker/Makefile.am	2015-03-19 12:40:17.784812000 -0500
@@ -3,7 +3,7 @@ EXTRA_DIST = \
 
 plugindir = $(libdir)/pidgin
 
-ticker_la_LDFLAGS = -module -avoid-version
+ticker_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
