--- origsrc/pidgin-2.5.0/configure.ac	2008-08-18 20:53:21.000000000 -0500
+++ src/pidgin-2.5.0/configure.ac	2008-08-25 21:33:10.073090200 -0500
@@ -119,6 +119,30 @@
 GETTEXT_PACKAGE=pidgin
 AC_SUBST(GETTEXT_PACKAGE)
 
+# win32
+case "$host_os" in
+cygwin*)
+	platform_win32=yes
+	LIBFINCH="-lfinch"
+	LIBPIDGIN="-lpidgin"
+	FINCH_PLUGIN_LDFLAGS="-no-undefined -Wl,\$(top_builddir)/finch/libfinch.a"
+	PIDGIN_PLUGIN_LDFLAGS="-no-undefined -Wl,\$(top_builddir)/pidgin/libpidgin.a"
+	;;
+*)
+	platform_win32=no
+	LIBFINCH=
+	LIBPIDGIN=
+	LIBPIDGIN=
+	FINCH_PLUGIN_LDFLAGS=
+	PIDGIN_PLUGIN_LDFLAGS=
+	;;
+esac
+
+AM_CONDITIONAL(PLATFORM_WIN32, test "x$platform_win32" = "xyes")
+AC_SUBST(LIBFINCH)
+AC_SUBST(LIBPIDGIN)
+AC_SUBST(FINCH_PLUGIN_LDFLAGS)
+AC_SUBST(PIDGIN_PLUGIN_LDFLAGS)
 
 # before gettexting, in case iconv matters
 case "$host_os" in
--- origsrc/pidgin-2.5.0/finch/Makefile.am	2008-08-18 20:53:22.000000000 -0500
+++ src/pidgin-2.5.0/finch/Makefile.am	2008-08-25 21:33:53.962310700 -0500
@@ -7,7 +7,7 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = finch.pc
 
-SUBDIRS = libgnt plugins
+SUBDIRS = libgnt . plugins
 
 if ENABLE_GNT
 
@@ -62,7 +62,13 @@
 	$(finch_headers)
 
 finch_DEPENDENCIES = @LIBOBJS@
+
+if PLATFORM_WIN32
+finch_LDFLAGS = -Wl,--export-all-symbols -Wl,--out-implib,libfinch.a
+else
 finch_LDFLAGS = -export-dynamic
+endif
+
 finch_LDADD = \
 	@LIBOBJS@ \
 	$(DBUS_LIBS) \
@@ -90,3 +96,9 @@
 	$(LIBXML_CFLAGS) \
 	$(GSTREAMER_CFLAGS) \
 	$(GNT_CFLAGS)
+
+if PLATFORM_WIN32
+install-data-local:
+	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) libfinch.a $(DESTDIR)$(libdir)
+endif
--- origsrc/pidgin-2.5.0/finch/finch.pc.in	2008-08-18 20:53:22.000000000 -0500
+++ src/pidgin-2.5.0/finch/finch.pc.in	2008-08-25 21:33:53.977935200 -0500
@@ -11,4 +11,5 @@
 Version: @VERSION@
 Requires: gnt purple
 Cflags: -I${includedir}/finch
+Libs: -L${libdir} @LIBFINCH@
 
--- origsrc/pidgin-2.5.0/finch/libgnt/wms/Makefile.am	2008-08-18 20:53:28.000000000 -0500
+++ src/pidgin-2.5.0/finch/libgnt/wms/Makefile.am	2008-08-25 21:33:53.993559700 -0500
@@ -5,8 +5,8 @@
 purple_wms =
 endif
 
-s_la_LDFLAGS             = -module -avoid-version
-irssi_la_LDFLAGS         = -module -avoid-version
+s_la_LDFLAGS             = -module -avoid-version -no-undefined
+irssi_la_LDFLAGS         = -module -avoid-version -no-undefined
 
 plugin_LTLIBRARIES = \
 	$(purple_wms)
--- origsrc/pidgin-2.5.0/finch/plugins/Makefile.am	2008-08-18 20:53:28.000000000 -0500
+++ src/pidgin-2.5.0/finch/plugins/Makefile.am	2008-08-25 21:33:53.993559700 -0500
@@ -1,8 +1,8 @@
-gntclipboard_la_LDFLAGS = -module -avoid-version
-gntgf_la_LDFLAGS      = -module -avoid-version
-gnthistory_la_LDFLAGS = -module -avoid-version
-gntlastlog_la_LDFLAGS = -module -avoid-version
-grouping_la_LDFLAGS   = -module -avoid-version
+gntclipboard_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gntgf_la_LDFLAGS      = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gnthistory_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gntlastlog_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+grouping_la_LDFLAGS   = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -24,11 +24,11 @@
 gntclipboard_la_CFLAGS = $(X11_CFLAGS)
 gntgf_la_CFLAGS = $(X11_CFLAGS)
 
-gntclipboard_la_LIBADD = $(GLIB_LIBS) $(X11_LIBS)
-gntgf_la_LIBADD       = $(GLIB_LIBS) $(X11_LIBS) $(top_builddir)/finch/libgnt/libgnt.la
-gnthistory_la_LIBADD  = $(GLIB_LIBS)
-gntlastlog_la_LIBADD  = $(GLIB_LIBS)
-grouping_la_LIBADD    = $(GLIB_LIBS) $(top_builddir)/finch/libgnt/libgnt.la
+gntclipboard_la_LIBADD = $(GLIB_LIBS) $(X11_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
+gntgf_la_LIBADD       = $(GLIB_LIBS) $(X11_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
+gnthistory_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
+gntlastlog_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
+grouping_la_LIBADD    = $(GLIB_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
 
 endif # PLUGINS
 
--- origsrc/pidgin-2.5.0/libpurple/Makefile.am	2008-08-18 20:53:28.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/Makefile.am	2008-08-25 21:33:54.009184200 -0500
@@ -30,7 +30,7 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = purple.pc
 
-SUBDIRS = $(GCONF_DIR) plugins protocols tests . example
+SUBDIRS = . $(GCONF_DIR) plugins protocols tests example
 
 purple_coresources = \
 	account.c \
--- origsrc/pidgin-2.5.0/libpurple/plugins/Makefile.am	2008-08-18 20:53:32.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/plugins/Makefile.am	2008-08-25 21:33:54.024808700 -0500
@@ -24,26 +24,26 @@
 
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-autoaccept_la_LDFLAGS       = -module -avoid-version
-buddynote_la_LDFLAGS        = -module -avoid-version
-ciphertest_la_LDFLAGS		= -module -avoid-version
-codeinline_la_LDFLAGS		= -module -avoid-version
-debug_example_la_LDFLAGS    = -module -avoid-version
-helloworld_la_LDFLAGS       = -module -avoid-version
-idle_la_LDFLAGS             = -module -avoid-version
-joinpart_la_LDFLAGS         = -module -avoid-version
-log_reader_la_LDFLAGS       = -module -avoid-version
-newline_la_LDFLAGS          = -module -avoid-version
-notify_example_la_LDFLAGS   = -module -avoid-version
-offlinemsg_la_LDFLAGS       = -module -avoid-version
-pluginpref_example_la_LDFLAGS = -module -avoid-version
-psychic_la_LDFLAGS          = -module -avoid-version
-signals_test_la_LDFLAGS		= -module -avoid-version
-simple_la_LDFLAGS			= -module -avoid-version
-statenotify_la_LDFLAGS      = -module -avoid-version
+autoaccept_la_LDFLAGS       = -module -avoid-version -no-undefined
+buddynote_la_LDFLAGS        = -module -avoid-version -no-undefined
+ciphertest_la_LDFLAGS		= -module -avoid-version -no-undefined
+codeinline_la_LDFLAGS		= -module -avoid-version -no-undefined
+debug_example_la_LDFLAGS    = -module -avoid-version -no-undefined
+helloworld_la_LDFLAGS       = -module -avoid-version -no-undefined
+idle_la_LDFLAGS             = -module -avoid-version -no-undefined
+joinpart_la_LDFLAGS         = -module -avoid-version -no-undefined
+log_reader_la_LDFLAGS       = -module -avoid-version -no-undefined
+newline_la_LDFLAGS          = -module -avoid-version -no-undefined
+notify_example_la_LDFLAGS   = -module -avoid-version -no-undefined
+offlinemsg_la_LDFLAGS       = -module -avoid-version -no-undefined
+pluginpref_example_la_LDFLAGS = -module -avoid-version -no-undefined
+psychic_la_LDFLAGS          = -module -avoid-version -no-undefined
+signals_test_la_LDFLAGS		= -module -avoid-version -no-undefined
+simple_la_LDFLAGS			= -module -avoid-version -no-undefined
+statenotify_la_LDFLAGS      = -module -avoid-version -no-undefined
 
 # this can't be in a conditional otherwise automake 1.4 yells
-dbus_example_la_LDFLAGS     = -module -avoid-version
+dbus_example_la_LDFLAGS     = -module -avoid-version -no-undefined
 
 if PLUGINS
 
@@ -87,28 +87,28 @@
 simple_la_SOURCES			= simple.c
 statenotify_la_SOURCES      = statenotify.c
 
-autoaccept_la_LIBADD        = $(GLIB_LIBS)
-buddynote_la_LIBADD         = $(GLIB_LIBS)
-ciphertest_la_LIBADD		= $(GLIB_LIBS)
-codeinline_la_LIBADD		= $(GLIB_LIBS)
-idle_la_LIBADD              = $(GLIB_LIBS)
-joinpart_la_LIBADD          = $(GLIB_LIBS)
-log_reader_la_LIBADD        = $(GLIB_LIBS)
-newline_la_LIBADD           = $(GLIB_LIBS)
-notify_example_la_LIBADD    = $(GLIB_LIBS)
-offlinemsg_la_LIBADD        = $(GLIB_LIBS)
-pluginpref_example_la_LIBADD = $(GLIB_LIBS)
-psychic_la_LIBADD           = $(GLIB_LIBS)
-signals_test_la_LIBADD		= $(GLIB_LIBS)
-simple_la_LIBADD			= $(GLIB_LIBS)
-statenotify_la_LIBADD       = $(GLIB_LIBS)
+autoaccept_la_LIBADD        = ../libpurple.la $(GLIB_LIBS)
+buddynote_la_LIBADD         = ../libpurple.la $(GLIB_LIBS)
+ciphertest_la_LIBADD		= ../libpurple.la $(GLIB_LIBS)
+codeinline_la_LIBADD		= ../libpurple.la $(GLIB_LIBS)
+idle_la_LIBADD              = ../libpurple.la $(GLIB_LIBS)
+joinpart_la_LIBADD          = ../libpurple.la $(GLIB_LIBS)
+log_reader_la_LIBADD        = ../libpurple.la $(GLIB_LIBS)
+newline_la_LIBADD           = ../libpurple.la $(GLIB_LIBS)
+notify_example_la_LIBADD    = ../libpurple.la $(GLIB_LIBS)
+offlinemsg_la_LIBADD        = ../libpurple.la $(GLIB_LIBS)
+pluginpref_example_la_LIBADD = ../libpurple.la $(GLIB_LIBS)
+psychic_la_LIBADD           = ../libpurple.la $(GLIB_LIBS)
+signals_test_la_LIBADD		= ../libpurple.la $(GLIB_LIBS)
+simple_la_LIBADD			= ../libpurple.la $(GLIB_LIBS)
+statenotify_la_LIBADD       = ../libpurple.la $(GLIB_LIBS)
 
 if ENABLE_DBUS
 
 CLEANFILES              = dbus-example-bindings.c
 dbus_example_la_SOURCES = dbus-example.c
 
-dbus_example_la_LIBADD      = $(GLIB_LIBS) $(DBUS_LIBS)
+dbus_example_la_LIBADD      = ../libpurple.la $(GLIB_LIBS) $(DBUS_LIBS)
 
 .PHONY: always
 
--- origsrc/pidgin-2.5.0/libpurple/plugins/mono/loader/Makefile.am	2008-08-18 20:53:34.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/plugins/mono/loader/Makefile.am	2008-08-25 21:33:54.056057700 -0500
@@ -12,9 +12,9 @@
 	blist-glue.c \
 	status-glue.c 
 
-mono_la_LDFLAGS  = -module -avoid-version
+mono_la_LDFLAGS  = -module -avoid-version -no-undefined
 
-mono_la_LIBADD = $(MONO_LIBS)
+mono_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(MONO_LIBS)
 
 AM_CPPFLAGS = \
 	-I$(top_srcdir) \
--- origsrc/pidgin-2.5.0/libpurple/plugins/perl/Makefile.am	2008-08-18 20:53:34.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/plugins/perl/Makefile.am	2008-08-25 21:33:54.071682200 -0500
@@ -4,8 +4,8 @@
 
 plugin_LTLIBRARIES = perl.la
 
-perl_la_LDFLAGS = -module -avoid-version
-perl_la_LIBADD = $(GLIB_LIBS) $(PERL_LIBS)
+perl_la_LDFLAGS = -module -avoid-version -no-undefined
+perl_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(PERL_LIBS)
 perl_la_SOURCES = \
 	perl.c \
 	perl-common.c \
--- origsrc/pidgin-2.5.0/libpurple/plugins/perl/common/Makefile.PL.in	2008-08-18 20:53:34.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/plugins/perl/common/Makefile.PL.in	2008-08-25 21:33:54.087306700 -0500
@@ -11,6 +11,7 @@
        AUTHOR           => 'Purple <http://pidgin.im/>') : ()),
     'DEFINE'            => '@DEBUG_CFLAGS@',
     'INC'               => '-I. -I@srcdir@ -I@top_srcdir@ -I@top_srcdir@/libpurple @GLIB_CFLAGS@',
+    'LIBS'              => [ '-L@top_builddir@/libpurple/.libs -lpurple @GLIB_LIBS@' ],
     'OBJECT'            => '$(O_FILES)', # link all the C files too
 #    'OPTIMIZE'          => '-g', # For debugging
 );
--- origsrc/pidgin-2.5.0/libpurple/plugins/ssl/Makefile.am	2008-08-18 20:53:35.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/plugins/ssl/Makefile.am	2008-08-25 21:33:54.087306700 -0500
@@ -3,9 +3,9 @@
 
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-ssl_la_LDFLAGS        = -module -avoid-version
-ssl_gnutls_la_LDFLAGS = -module -avoid-version
-ssl_nss_la_LDFLAGS    = -module -avoid-version
+ssl_la_LDFLAGS        = -module -avoid-version -no-undefined
+ssl_gnutls_la_LDFLAGS = -module -avoid-version -no-undefined
+ssl_nss_la_LDFLAGS    = -module -avoid-version -no-undefined
 
 if PLUGINS
 
@@ -18,9 +18,9 @@
 ssl_gnutls_la_SOURCES = ssl-gnutls.c
 ssl_nss_la_SOURCES    = ssl-nss.c
 
-ssl_la_LIBADD        = $(GLIB_LIBS)
-ssl_gnutls_la_LIBADD = $(GLIB_LIBS) $(GNUTLS_LIBS)
-ssl_nss_la_LIBADD    = $(GLIB_LIBS) $(NSS_LIBS)
+ssl_la_LIBADD        = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
+ssl_gnutls_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(GNUTLS_LIBS)
+ssl_nss_la_LIBADD    = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(NSS_LIBS)
 
 endif # PLUGINS
 
--- origsrc/pidgin-2.5.0/libpurple/plugins/tcl/Makefile.am	2008-08-18 20:53:35.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/plugins/tcl/Makefile.am	2008-08-25 21:33:54.118555700 -0500
@@ -1,13 +1,13 @@
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-tcl_la_LDFLAGS = -module -avoid-version
+tcl_la_LDFLAGS = -module -avoid-version -no-undefined
 
 plugin_LTLIBRARIES = tcl.la
 
 tcl_la_SOURCES = tcl.c tcl_glib.c tcl_glib.h tcl_cmds.c tcl_signals.c tcl_purple.h \
                  tcl_ref.c tcl_cmd.c
 
-tcl_la_LIBADD = $(GLIB_LIBS) $(TCL_LIBS) $(TK_LIBS)
+tcl_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(TCL_LIBS) $(TK_LIBS)
 
 EXTRA_DIST = signal-test.tcl Makefile.mingw
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/bonjour/Makefile.am	2008-08-18 20:53:36.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/bonjour/Makefile.am	2008-08-25 21:33:54.134180200 -0500
@@ -25,7 +25,7 @@
 
 AM_CFLAGS = $(st)
 
-libbonjour_la_LDFLAGS = -module -avoid-version
+libbonjour_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_BONJOUR
 
@@ -40,7 +40,7 @@
 st =
 pkg_LTLIBRARIES       = libbonjour.la
 libbonjour_la_SOURCES = $(BONJOURSOURCES)
-libbonjour_la_LIBADD  = $(GLIB_LIBS) $(LIBXML_LIBS) $(AVAHI_LIBS)
+libbonjour_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(LIBXML_LIBS) $(AVAHI_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/gg/Makefile.am	2008-08-18 20:53:37.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/gg/Makefile.am	2008-08-25 21:33:54.149804700 -0500
@@ -47,7 +47,7 @@
 
 AM_CFLAGS = $(st)
 
-libgg_la_LDFLAGS = -module -avoid-version
+libgg_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_GG
 
@@ -66,7 +66,7 @@
 noinst_LIBRARIES =
 
 libgg_la_SOURCES = $(GGSOURCES)
-libgg_la_LIBADD  = $(GLIB_LIBS) $(GADU_LIBS)
+libgg_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(GADU_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/irc/Makefile.am	2008-08-18 20:53:38.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/irc/Makefile.am	2008-08-25 21:33:54.149804700 -0500
@@ -6,7 +6,7 @@
 
 AM_CFLAGS = $(st)
 
-libirc_la_LDFLAGS = -module -avoid-version
+libirc_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_IRC
 
@@ -24,7 +24,7 @@
 noinst_LIBRARIES =
 
 libirc_la_SOURCES = $(IRCSOURCES)
-libirc_la_LIBADD  = $(GLIB_LIBS)
+libirc_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/jabber/Makefile.am	2008-08-18 20:53:38.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/jabber/Makefile.am	2008-08-25 21:33:54.181053700 -0500
@@ -52,7 +52,7 @@
 
 AM_CFLAGS = $(st)
 
-libxmpp_la_LDFLAGS = -module -avoid-version
+libxmpp_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_JABBER
 
@@ -66,14 +66,14 @@
 else
 
 st =
-pkg_LTLIBRARIES = libjabber.la libxmpp.la
-noinst_LIBRARIES =
+pkg_LTLIBRARIES = libxmpp.la
+noinst_LTLIBRARIES = libjabber.la
 
 libjabber_la_SOURCES = $(JABBERSOURCES)
 libjabber_la_LIBADD = $(GLIB_LIBS) $(SASL_LIBS) $(LIBXML_LIBS)
 
 libxmpp_la_SOURCES = libxmpp.c
-libxmpp_la_LIBADD = libjabber.la
+libxmpp_la_LIBADD = libjabber.la $(top_builddir)/libpurple/libpurple.la
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/msn/Makefile.am	2008-08-18 20:53:40.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/msn/Makefile.am	2008-08-25 21:33:54.196678200 -0500
@@ -69,7 +69,7 @@
 
 AM_CFLAGS = $(st)
 
-libmsn_la_LDFLAGS = -module -avoid-version
+libmsn_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_MSN
 
@@ -83,7 +83,7 @@
 st =
 pkg_LTLIBRARIES   = libmsn.la
 libmsn_la_SOURCES = $(MSNSOURCES)
-libmsn_la_LIBADD  = $(GLIB_LIBS)
+libmsn_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/msnp9/Makefile.am	2008-08-18 20:53:42.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/msnp9/Makefile.am	2008-08-25 21:33:54.212302700 -0500
@@ -65,7 +65,7 @@
 
 AM_CFLAGS = $(st)
 
-libmsn_la_LDFLAGS = -module -avoid-version
+libmsn_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_MSN
 
@@ -79,7 +79,7 @@
 st =
 pkg_LTLIBRARIES   = libmsn.la
 libmsn_la_SOURCES = $(MSNP9SOURCES)
-libmsn_la_LIBADD  = $(GLIB_LIBS)
+libmsn_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/myspace/Makefile.am	2008-08-18 20:53:44.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/myspace/Makefile.am	2008-08-25 21:33:54.243551700 -0500
@@ -18,7 +18,7 @@
 
 AM_CFLAGS = $(st)
 
-libmyspace_la_LDFLAGS = -module -avoid-version
+libmyspace_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_MYSPACE
 
@@ -32,7 +32,7 @@
 st =
 pkg_LTLIBRARIES       = libmyspace.la
 libmyspace_la_SOURCES = $(MSIMSOURCES)
-libmyspace_la_LIBADD  = $(GLIB_LIBS)
+libmyspace_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/novell/Makefile.am	2008-08-18 20:53:44.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/novell/Makefile.am	2008-08-25 21:33:54.259176200 -0500
@@ -28,7 +28,7 @@
 
 AM_CFLAGS = $(st)
 
-libnovell_la_LDFLAGS = -module -avoid-version
+libnovell_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_NOVELL
 
@@ -46,7 +46,7 @@
 noinst_LIBRARIES =
 
 libnovell_la_SOURCES = $(NOVELLSOURCES)
-libnovell_la_LIBADD  = $(GLIB_LIBS)
+libnovell_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/null/Makefile.am	2008-08-18 20:53:44.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/null/Makefile.am	2008-08-25 21:33:54.274800700 -0500
@@ -6,13 +6,13 @@
 
 AM_CFLAGS = $(st)
 
-libnull_la_LDFLAGS = -module -avoid-version
+libnull_la_LDFLAGS = -module -avoid-version -no-undefined
 
 # nullprpl isn't built by default. when it is built, it's dynamically linked.
 st =
 pkg_LTLIBRARIES   = libnull.la
 libnull_la_SOURCES = $(NULLSOURCES)
-libnull_la_LIBADD  = $(GLIB_LIBS)
+libnull_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/libpurple \
--- origsrc/pidgin-2.5.0/libpurple/protocols/oscar/Makefile.am	2008-08-18 20:53:45.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/oscar/Makefile.am	2008-08-25 21:33:54.274800700 -0500
@@ -47,8 +47,8 @@
 
 AM_CFLAGS = $(st)
 
-libaim_la_LDFLAGS = -module -avoid-version
-libicq_la_LDFLAGS = -module -avoid-version
+libaim_la_LDFLAGS = -module -avoid-version -no-undefined
+libicq_la_LDFLAGS = -module -avoid-version -no-undefined
 if STATIC_OSCAR
 
 st = -DPURPLE_STATIC_PRPL
@@ -60,9 +60,10 @@
 else
 
 st =
-pkg_LTLIBRARIES     = liboscar.la libaim.la libicq.la
+noinst_LTLIBRARIES = liboscar.la
+pkg_LTLIBRARIES     = libaim.la libicq.la
 liboscar_la_SOURCES = $(OSCARSOURCES)
-liboscar_la_LIBADD  = $(GLIB_LIBS)
+liboscar_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 libaim_la_SOURCES = libaim.c
 libaim_la_LIBADD = liboscar.la
--- origsrc/pidgin-2.5.0/libpurple/protocols/qq/Makefile.am	2008-08-18 20:53:46.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/qq/Makefile.am	2008-08-25 21:33:54.337298700 -0500
@@ -61,7 +61,7 @@
 
 AM_CFLAGS = $(st)
 
-libqq_la_LDFLAGS = -module -avoid-version
+libqq_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_QQ
 
@@ -75,7 +75,7 @@
 st =
 pkg_LTLIBRARIES  = libqq.la
 libqq_la_SOURCES = $(QQSOURCES)
-libqq_la_LIBADD  = $(GLIB_LIBS)
+libqq_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/sametime/Makefile.am	2008-08-18 20:53:47.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/sametime/Makefile.am	2008-08-25 21:33:54.337298700 -0500
@@ -27,8 +27,8 @@
 endif
 
 
-libsametime_la_LDFLAGS = -module -avoid-version
-libsametime_la_LIBADD = $(GLIB_LIBS) $(MEANWHILE_LIBS)
+libsametime_la_LDFLAGS = -module -avoid-version -no-undefined
+libsametime_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(MEANWHILE_LIBS)
 
 
 AM_CFLAGS = \
--- origsrc/pidgin-2.5.0/libpurple/protocols/silc/Makefile.am	2008-08-18 20:53:47.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/silc/Makefile.am	2008-08-25 21:33:54.352923200 -0500
@@ -6,7 +6,7 @@
 
 AM_CFLAGS = $(st)
 
-libsilcpurple_la_LDFLAGS = -module -avoid-version
+libsilcpurple_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_SILC
 
@@ -25,7 +25,7 @@
 noinst_LIBRARIES =
 
 libsilcpurple_la_SOURCES = $(SILCSOURCES)
-libsilcpurple_la_LIBADD  = $(GLIB_LIBS) $(SILC_LIBS)
+libsilcpurple_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(SILC_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/silc10/Makefile.am	2008-08-18 20:53:47.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/silc10/Makefile.am	2008-08-25 21:33:54.368547700 -0500
@@ -6,7 +6,7 @@
 
 AM_CFLAGS = $(st)
 
-libsilcpurple_la_LDFLAGS = -module -avoid-version
+libsilcpurple_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_SILC
 
@@ -25,7 +25,7 @@
 noinst_LIBRARIES =
 
 libsilcpurple_la_SOURCES = $(SILCSOURCES)
-libsilcpurple_la_LIBADD  = $(GLIB_LIBS) $(SILC_LIBS)
+libsilcpurple_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(SILC_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/simple/Makefile.am	2008-08-18 20:53:48.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/simple/Makefile.am	2008-08-25 21:33:54.415421200 -0500
@@ -11,7 +11,7 @@
 
 AM_CFLAGS = $(st) 
 
-libsimple_la_LDFLAGS = -module -avoid-version
+libsimple_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_MSN
 
@@ -25,7 +25,7 @@
 st =
 pkg_LTLIBRARIES   = libsimple.la
 libsimple_la_SOURCES = $(SIMPLESOURCES)
-libsimple_la_LIBADD  = $(GLIB_LIBS)
+libsimple_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/toc/Makefile.am	2008-08-18 20:53:48.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/toc/Makefile.am	2008-08-25 21:33:54.431045700 -0500
@@ -8,7 +8,7 @@
 
 AM_CFLAGS = $(st)
 
-libtoc_la_LDFLAGS = -module -avoid-version
+libtoc_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_TOC
 
@@ -22,6 +22,7 @@
 st =
 pkg_LTLIBRARIES   = libtoc.la
 libtoc_la_SOURCES = $(TOCSOURCES)
+libtoc_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/yahoo/Makefile.am	2008-08-18 20:53:48.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/yahoo/Makefile.am	2008-08-25 21:33:54.431045700 -0500
@@ -31,7 +31,7 @@
 
 AM_CFLAGS = $(st)
 
-libyahoo_la_LDFLAGS = -module -avoid-version
+libyahoo_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_YAHOO
 
@@ -45,7 +45,7 @@
 st =
 pkg_LTLIBRARIES     = libyahoo.la
 libyahoo_la_SOURCES = $(YAHOOSOURCES)
-libyahoo_la_LIBADD  = $(GLIB_LIBS)
+libyahoo_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/libpurple/protocols/zephyr/Makefile.am	2008-08-18 20:53:49.000000000 -0500
+++ src/pidgin-2.5.0/libpurple/protocols/zephyr/Makefile.am	2008-08-25 21:33:54.477919200 -0500
@@ -73,7 +73,7 @@
 
 ZEPHYRLIBSEXT = $(ZEPHYR_LDFLAGS) $(ZEPHYR_LIBS)
 
-libzephyr_la_LDFLAGS = -module -avoid-version
+libzephyr_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_ZEPHYR
 
@@ -90,10 +90,10 @@
 
 if EXTERNAL_LIBZEPHYR
 libzephyr_la_SOURCES = $(ZEPHYRSOURCESEXT)
-libzephyr_la_LIBADD = $(GLIB_LIBS) $(ZEPHYRLIBSEXT)
+libzephyr_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(ZEPHYRLIBSEXT)
 else
 libzephyr_la_SOURCES = $(ZEPHYRSOURCES)
-libzephyr_la_LIBADD  = $(GLIB_LIBS) $(ZEPHYRLIBS)
+libzephyr_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(ZEPHYRLIBS)
 endif
 
 
--- origsrc/pidgin-2.5.0/pidgin/Makefile.am	2008-08-18 20:53:51.000000000 -0500
+++ src/pidgin-2.5.0/pidgin/Makefile.am	2008-08-25 21:33:54.477919200 -0500
@@ -68,7 +68,7 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = pidgin.pc
 
-SUBDIRS = pixmaps plugins
+SUBDIRS = . pixmaps plugins
 
 bin_PROGRAMS = pidgin
 
@@ -184,7 +184,13 @@
 
 
 pidgin_DEPENDENCIES = @LIBOBJS@
+
+if PLATFORM_WIN32
+pidgin_LDFLAGS = -Wl,--export-all-symbols,--out-implib,libpidgin.a
+else
 pidgin_LDFLAGS = -export-dynamic
+endif
+
 pidgin_LDADD = \
 	@LIBOBJS@ \
 	$(DBUS_LIBS) \
@@ -222,3 +228,9 @@
 	$(LIBXML_CFLAGS) \
 	$(INTGG_CFLAGS)
 endif  # ENABLE_GTK
+
+if PLATFORM_WIN32
+install-data-local:
+	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) libpidgin.a $(DESTDIR)$(libdir)
+endif
--- origsrc/pidgin-2.5.0/pidgin/pidgin-uninstalled.pc.in	2008-08-18 20:53:53.000000000 -0500
+++ src/pidgin-2.5.0/pidgin/pidgin-uninstalled.pc.in	2008-08-25 21:33:54.493543700 -0500
@@ -11,3 +11,4 @@
 Version: @VERSION@
 Requires: gtk+-2.0 purple
 Cflags: -I${pcfiledir}
+Libs: -L${pcfiledir} @LIBPIDGIN@
--- origsrc/pidgin-2.5.0/pidgin/pidgin.pc.in	2008-08-18 20:53:53.000000000 -0500
+++ src/pidgin-2.5.0/pidgin/pidgin.pc.in	2008-08-25 21:33:54.509168200 -0500
@@ -11,4 +11,5 @@
 Version: @VERSION@
 Requires: gtk+-2.0 purple
 Cflags: -I${includedir}/pidgin
+Libs: -L${libdir} @LIBPIDGIN@
 
--- origsrc/pidgin-2.5.0/pidgin/plugins/Makefile.am	2008-08-18 20:54:02.000000000 -0500
+++ src/pidgin-2.5.0/pidgin/plugins/Makefile.am	2008-08-25 21:33:54.509168200 -0500
@@ -30,22 +30,22 @@
 
 plugindir = $(libdir)/pidgin
 
-convcolors_la_LDFLAGS       = -module -avoid-version
-contact_priority_la_LDFLAGS = -module -avoid-version
-extplacement_la_LDFLAGS     = -module -avoid-version
-gtk_signals_test_la_LDFLAGS = -module -avoid-version
-gtkbuddynote_la_LDFLAGS     = -module -avoid-version
-history_la_LDFLAGS          = -module -avoid-version
-iconaway_la_LDFLAGS         = -module -avoid-version
-markerline_la_LDFLAGS       = -module -avoid-version
-notify_la_LDFLAGS           = -module -avoid-version
-pidginrc_la_LDFLAGS         = -module -avoid-version
-relnot_la_LDFLAGS           = -module -avoid-version
-sendbutton_la_LDFLAGS       = -module -avoid-version
-spellchk_la_LDFLAGS         = -module -avoid-version
-timestamp_la_LDFLAGS        = -module -avoid-version
-timestamp_format_la_LDFLAGS = -module -avoid-version
-xmppconsole_la_LDFLAGS      = -module -avoid-version
+convcolors_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+contact_priority_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+extplacement_la_LDFLAGS     = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+gtk_signals_test_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+gtkbuddynote_la_LDFLAGS     = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+history_la_LDFLAGS          = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+iconaway_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+markerline_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+notify_la_LDFLAGS           = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+pidginrc_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+relnot_la_LDFLAGS           = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+sendbutton_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+spellchk_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+timestamp_la_LDFLAGS        = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+timestamp_format_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+xmppconsole_la_LDFLAGS      = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -86,22 +86,22 @@
 timestamp_format_la_SOURCES = timestamp_format.c
 xmppconsole_la_SOURCES      = xmppconsole.c
 
-convcolors_la_LIBADD        = $(GTK_LIBS)
-contact_priority_la_LIBADD  = $(GTK_LIBS)
-extplacement_la_LIBADD      = $(GTK_LIBS)
-gtk_signals_test_la_LIBADD  = $(GTK_LIBS)
-gtkbuddynote_la_LIBADD      = $(GTK_LIBS)
-history_la_LIBADD           = $(GTK_LIBS)
-iconaway_la_LIBADD          = $(GTK_LIBS)
-markerline_la_LIBADD        = $(GTK_LIBS)
-notify_la_LIBADD            = $(GTK_LIBS)
-pidginrc_la_LIBADD          = $(GTK_LIBS)
-relnot_la_LIBADD            = $(GLIB_LIBS)
-sendbutton_la_LIBADD        = $(GTK_LIBS)
-spellchk_la_LIBADD          = $(GTK_LIBS)
-timestamp_la_LIBADD         = $(GTK_LIBS)
-timestamp_format_la_LIBADD  = $(GTK_LIBS)
-xmppconsole_la_LIBADD       = $(GTK_LIBS)
+convcolors_la_LIBADD        = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+contact_priority_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+extplacement_la_LIBADD      = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+gtk_signals_test_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+gtkbuddynote_la_LIBADD      = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+history_la_LIBADD           = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+iconaway_la_LIBADD          = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+markerline_la_LIBADD        = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+notify_la_LIBADD            = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+pidginrc_la_LIBADD          = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+relnot_la_LIBADD            = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
+sendbutton_la_LIBADD        = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+spellchk_la_LIBADD          = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+timestamp_la_LIBADD         = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+timestamp_format_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+xmppconsole_la_LIBADD       = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
 
 endif # PLUGINS
 
--- origsrc/pidgin-2.5.0/pidgin/plugins/cap/Makefile.am	2008-08-18 20:54:02.000000000 -0500
+++ src/pidgin-2.5.0/pidgin/plugins/cap/Makefile.am	2008-08-25 21:33:54.524792700 -0500
@@ -1,6 +1,6 @@
 capdir = $(libdir)/pidgin
 
-cap_la_LDFLAGS = -module -avoid-version
+cap_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -15,7 +15,7 @@
 
 endif
 
-cap_la_LIBADD = $(GTK_LIBS) $(SQLITE3_LIBS)
+cap_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS) $(SQLITE3_LIBS)
 
 AM_CPPFLAGS = \
 	-DDATADIR=\"$(datadir)\" \
--- origsrc/pidgin-2.5.0/pidgin/plugins/gestures/Makefile.am	2008-08-18 20:54:03.000000000 -0500
+++ src/pidgin-2.5.0/pidgin/plugins/gestures/Makefile.am	2008-08-25 21:33:54.571666200 -0500
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/pidgin
 
-gestures_la_LDFLAGS = -module -avoid-version
+gestures_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -13,7 +13,7 @@
 	stroke.c \
 	stroke-draw.c
 
-gestures_la_LIBADD = $(GTK_LIBS)
+gestures_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/pidgin/plugins/gevolution/Makefile.am	2008-08-18 20:54:03.000000000 -0500
+++ src/pidgin-2.5.0/pidgin/plugins/gevolution/Makefile.am	2008-08-25 21:33:54.571666200 -0500
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/pidgin
 
-gevolution_la_LDFLAGS = -module -avoid-version
+gevolution_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -15,7 +15,7 @@
 	new_person_dialog.c \
 	eds-utils.c
 
-gevolution_la_LIBADD = $(EVOLUTION_ADDRESSBOOK_LIBS) $(GTK_LIBS)
+gevolution_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(EVOLUTION_ADDRESSBOOK_LIBS) $(GTK_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.0/pidgin/plugins/musicmessaging/Makefile.am	2008-08-18 20:54:03.000000000 -0500
+++ src/pidgin-2.5.0/pidgin/plugins/musicmessaging/Makefile.am	2008-08-25 21:33:54.602915200 -0500
@@ -3,7 +3,7 @@
 
 musicmessagingdir = $(libdir)/pidgin
 
-musicmessaging_la_LDFLAGS = -module -avoid-version
+musicmessaging_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 if ENABLE_DBUS
@@ -16,7 +16,7 @@
 musicmessaging_la_SOURCES = \
 	musicmessaging.c
 
-musicmessaging_la_LIBADD = $(GTK_LIBS) $(DBUS_LIBS)
+musicmessaging_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS) $(DBUS_LIBS)
 
 CLEANFILES              = music-messaging-bindings.c
 
--- origsrc/pidgin-2.5.0/pidgin/plugins/perl/common/Makefile.PL.in	2008-08-18 20:54:03.000000000 -0500
+++ src/pidgin-2.5.0/pidgin/plugins/perl/common/Makefile.PL.in	2008-08-25 21:33:54.634164200 -0500
@@ -10,6 +10,7 @@
        'AUTHOR'        => 'Pidgin <http://pidgin.im/>') :  ()),
     'DEFINE'        => '@DEBUG_CFLAGS@',
     'INC'           => '-I. -I@srcdir@ -I@top_srcdir@ -I@top_srcdir@/libpurple -I@top_srcdir@/pidgin @GTK_CFLAGS@',
+    'LIBS'          => ['-L@top_builddir@/libpurple/.libs -lpurple -L@top_builddir@/pidgin @LIBPIDGIN@ @GLIB_LIBS@'],
     'OBJECT'        => '$(O_FILES)', # link all the C files too
     'TYPEMAPS'      => ["@top_srcdir@/libpurple/plugins/perl/common/typemap"],
 #    'OPTIMIZE'      => '-g', # For debugging.
--- origsrc/pidgin-2.5.0/pidgin/plugins/ticker/Makefile.am	2008-08-18 20:54:03.000000000 -0500
+++ src/pidgin-2.5.0/pidgin/plugins/ticker/Makefile.am	2008-08-25 21:33:54.649788700 -0500
@@ -3,7 +3,7 @@
 
 plugindir = $(libdir)/pidgin
 
-ticker_la_LDFLAGS = -module -avoid-version
+ticker_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -14,7 +14,7 @@
 	gtkticker.h \
 	ticker.c
 
-ticker_la_LIBADD = $(GTK_LIBS)
+ticker_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
 
 endif
 
