--- origsrc/pidgin-2.5.6/configure.ac	2009-05-19 12:06:37.000000000 -0500
+++ src/pidgin-2.5.6/configure.ac	2009-06-15 13:46:06.485014000 -0500
@@ -118,6 +118,30 @@ AC_FUNC_ALLOCA
 GETTEXT_PACKAGE=pidgin
 AC_SUBST(GETTEXT_PACKAGE)
 
+# win32
+case "$host_os" in
+cygwin*)
+	platform_win32=yes
+	LIBFINCH="-lfinch"
+	LIBPIDGIN="-lpidgin"
+	FINCH_PLUGIN_LDFLAGS="-no-undefined -Wl,\$(top_builddir)/finch/libfinch.a"
+	PIDGIN_PLUGIN_LDFLAGS="-no-undefined -Wl,\$(top_builddir)/pidgin/libpidgin.a"
+	;;
+*)
+	platform_win32=no
+	LIBFINCH=
+	LIBPIDGIN=
+	LIBPIDGIN=
+	FINCH_PLUGIN_LDFLAGS=
+	PIDGIN_PLUGIN_LDFLAGS=
+	;;
+esac
+
+AM_CONDITIONAL(PLATFORM_WIN32, test "x$platform_win32" = "xyes")
+AC_SUBST(LIBFINCH)
+AC_SUBST(LIBPIDGIN)
+AC_SUBST(FINCH_PLUGIN_LDFLAGS)
+AC_SUBST(PIDGIN_PLUGIN_LDFLAGS)
 
 # before gettexting, in case iconv matters
 case "$host_os" in
--- origsrc/pidgin-2.5.6/finch/Makefile.am	2009-04-28 13:09:22.000000000 -0500
+++ src/pidgin-2.5.6/finch/Makefile.am	2009-06-15 13:46:06.487014100 -0500
@@ -7,7 +7,7 @@ EXTRA_DIST = \
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = finch.pc
 
-SUBDIRS = libgnt plugins
+SUBDIRS = libgnt . plugins
 
 if ENABLE_GNT
 
@@ -62,7 +62,13 @@ finchinclude_HEADERS = \
 	$(finch_headers)
 
 finch_DEPENDENCIES = @LIBOBJS@
+
+if PLATFORM_WIN32
+finch_LDFLAGS = -Wl,--export-all-symbols -Wl,--out-implib,libfinch.a
+else
 finch_LDFLAGS = -export-dynamic
+endif
+
 finch_LDADD = \
 	@LIBOBJS@ \
 	$(DBUS_LIBS) \
@@ -90,3 +96,9 @@ AM_CPPFLAGS = \
 	$(LIBXML_CFLAGS) \
 	$(GSTREAMER_CFLAGS) \
 	$(GNT_CFLAGS)
+
+if PLATFORM_WIN32
+install-data-local:
+	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) libfinch.a $(DESTDIR)$(libdir)
+endif
--- origsrc/pidgin-2.5.6/finch/finch.pc.in	2009-04-28 13:09:22.000000000 -0500
+++ src/pidgin-2.5.6/finch/finch.pc.in	2009-06-15 13:46:06.489014200 -0500
@@ -11,4 +11,5 @@ Description: Finch is an instant messeng
 Version: @VERSION@
 Requires: gnt purple
 Cflags: -I${includedir}/finch
+Libs: -L${libdir} @LIBFINCH@
 
--- origsrc/pidgin-2.5.6/finch/libgnt/wms/Makefile.am	2009-04-28 13:09:23.000000000 -0500
+++ src/pidgin-2.5.6/finch/libgnt/wms/Makefile.am	2009-06-15 13:46:06.492014400 -0500
@@ -6,8 +6,8 @@ else
 purple_wms =
 endif
 
-s_la_LDFLAGS             = -module -avoid-version
-irssi_la_LDFLAGS         = -module -avoid-version
+s_la_LDFLAGS             = -module -avoid-version -no-undefined
+irssi_la_LDFLAGS         = -module -avoid-version -no-undefined
 
 plugin_LTLIBRARIES = \
 	$(wms) \
--- origsrc/pidgin-2.5.6/finch/plugins/Makefile.am	2009-04-28 13:09:23.000000000 -0500
+++ src/pidgin-2.5.6/finch/plugins/Makefile.am	2009-06-15 13:46:06.495014500 -0500
@@ -1,8 +1,8 @@
-gntclipboard_la_LDFLAGS = -module -avoid-version
-gntgf_la_LDFLAGS      = -module -avoid-version
-gnthistory_la_LDFLAGS = -module -avoid-version
-gntlastlog_la_LDFLAGS = -module -avoid-version
-grouping_la_LDFLAGS   = -module -avoid-version
+gntclipboard_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gntgf_la_LDFLAGS      = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gnthistory_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gntlastlog_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+grouping_la_LDFLAGS   = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -24,11 +24,11 @@ grouping_la_SOURCES   = grouping.c
 gntclipboard_la_CFLAGS = $(X11_CFLAGS)
 gntgf_la_CFLAGS = $(X11_CFLAGS)
 
-gntclipboard_la_LIBADD = $(GLIB_LIBS) $(X11_LIBS)
-gntgf_la_LIBADD       = $(GLIB_LIBS) $(X11_LIBS) $(top_builddir)/finch/libgnt/libgnt.la
-gnthistory_la_LIBADD  = $(GLIB_LIBS)
-gntlastlog_la_LIBADD  = $(GLIB_LIBS)
-grouping_la_LIBADD    = $(GLIB_LIBS) $(top_builddir)/finch/libgnt/libgnt.la
+gntclipboard_la_LIBADD = $(GLIB_LIBS) $(X11_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
+gntgf_la_LIBADD       = $(GLIB_LIBS) $(X11_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
+gnthistory_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
+gntlastlog_la_LIBADD  = $(GLIB_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
+grouping_la_LIBADD    = $(GLIB_LIBS) $(top_builddir)/finch/libgnt/libgnt.la $(top_builddir)/libpurple/libpurple.la
 
 endif # PLUGINS
 
--- origsrc/pidgin-2.5.6/libpurple/Makefile.am	2009-04-28 13:09:23.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/Makefile.am	2009-06-15 13:46:06.497014600 -0500
@@ -30,7 +30,7 @@ endif
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = purple.pc
 
-SUBDIRS = $(GCONF_DIR) plugins protocols tests . example
+SUBDIRS = . $(GCONF_DIR) plugins protocols tests example
 
 purple_coresources = \
 	account.c \
--- origsrc/pidgin-2.5.6/libpurple/plugins/Makefile.am	2009-04-28 13:09:24.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/plugins/Makefile.am	2009-06-15 13:46:06.500014800 -0500
@@ -24,26 +24,26 @@ SUBDIRS = \
 
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-autoaccept_la_LDFLAGS       = -module -avoid-version
-buddynote_la_LDFLAGS        = -module -avoid-version
-ciphertest_la_LDFLAGS		= -module -avoid-version
-codeinline_la_LDFLAGS		= -module -avoid-version
-debug_example_la_LDFLAGS    = -module -avoid-version
-helloworld_la_LDFLAGS       = -module -avoid-version
-idle_la_LDFLAGS             = -module -avoid-version
-joinpart_la_LDFLAGS         = -module -avoid-version
-log_reader_la_LDFLAGS       = -module -avoid-version
-newline_la_LDFLAGS          = -module -avoid-version
-notify_example_la_LDFLAGS   = -module -avoid-version
-offlinemsg_la_LDFLAGS       = -module -avoid-version
-pluginpref_example_la_LDFLAGS = -module -avoid-version
-psychic_la_LDFLAGS          = -module -avoid-version
-signals_test_la_LDFLAGS		= -module -avoid-version
-simple_la_LDFLAGS			= -module -avoid-version
-statenotify_la_LDFLAGS      = -module -avoid-version
+autoaccept_la_LDFLAGS       = -module -avoid-version -no-undefined
+buddynote_la_LDFLAGS        = -module -avoid-version -no-undefined
+ciphertest_la_LDFLAGS		= -module -avoid-version -no-undefined
+codeinline_la_LDFLAGS		= -module -avoid-version -no-undefined
+debug_example_la_LDFLAGS    = -module -avoid-version -no-undefined
+helloworld_la_LDFLAGS       = -module -avoid-version -no-undefined
+idle_la_LDFLAGS             = -module -avoid-version -no-undefined
+joinpart_la_LDFLAGS         = -module -avoid-version -no-undefined
+log_reader_la_LDFLAGS       = -module -avoid-version -no-undefined
+newline_la_LDFLAGS          = -module -avoid-version -no-undefined
+notify_example_la_LDFLAGS   = -module -avoid-version -no-undefined
+offlinemsg_la_LDFLAGS       = -module -avoid-version -no-undefined
+pluginpref_example_la_LDFLAGS = -module -avoid-version -no-undefined
+psychic_la_LDFLAGS          = -module -avoid-version -no-undefined
+signals_test_la_LDFLAGS		= -module -avoid-version -no-undefined
+simple_la_LDFLAGS			= -module -avoid-version -no-undefined
+statenotify_la_LDFLAGS      = -module -avoid-version -no-undefined
 
 # this can't be in a conditional otherwise automake 1.4 yells
-dbus_example_la_LDFLAGS     = -module -avoid-version
+dbus_example_la_LDFLAGS     = -module -avoid-version -no-undefined
 
 if PLUGINS
 
@@ -87,28 +87,28 @@ signals_test_la_SOURCES		= signals-test.
 simple_la_SOURCES			= simple.c
 statenotify_la_SOURCES      = statenotify.c
 
-autoaccept_la_LIBADD        = $(GLIB_LIBS)
-buddynote_la_LIBADD         = $(GLIB_LIBS)
-ciphertest_la_LIBADD		= $(GLIB_LIBS)
-codeinline_la_LIBADD		= $(GLIB_LIBS)
-idle_la_LIBADD              = $(GLIB_LIBS)
-joinpart_la_LIBADD          = $(GLIB_LIBS)
-log_reader_la_LIBADD        = $(GLIB_LIBS)
-newline_la_LIBADD           = $(GLIB_LIBS)
-notify_example_la_LIBADD    = $(GLIB_LIBS)
-offlinemsg_la_LIBADD        = $(GLIB_LIBS)
-pluginpref_example_la_LIBADD = $(GLIB_LIBS)
-psychic_la_LIBADD           = $(GLIB_LIBS)
-signals_test_la_LIBADD		= $(GLIB_LIBS)
-simple_la_LIBADD			= $(GLIB_LIBS)
-statenotify_la_LIBADD       = $(GLIB_LIBS)
+autoaccept_la_LIBADD        = ../libpurple.la $(GLIB_LIBS)
+buddynote_la_LIBADD         = ../libpurple.la $(GLIB_LIBS)
+ciphertest_la_LIBADD		= ../libpurple.la $(GLIB_LIBS)
+codeinline_la_LIBADD		= ../libpurple.la $(GLIB_LIBS)
+idle_la_LIBADD              = ../libpurple.la $(GLIB_LIBS)
+joinpart_la_LIBADD          = ../libpurple.la $(GLIB_LIBS)
+log_reader_la_LIBADD        = ../libpurple.la $(GLIB_LIBS)
+newline_la_LIBADD           = ../libpurple.la $(GLIB_LIBS)
+notify_example_la_LIBADD    = ../libpurple.la $(GLIB_LIBS)
+offlinemsg_la_LIBADD        = ../libpurple.la $(GLIB_LIBS)
+pluginpref_example_la_LIBADD = ../libpurple.la $(GLIB_LIBS)
+psychic_la_LIBADD           = ../libpurple.la $(GLIB_LIBS)
+signals_test_la_LIBADD		= ../libpurple.la $(GLIB_LIBS)
+simple_la_LIBADD			= ../libpurple.la $(GLIB_LIBS)
+statenotify_la_LIBADD       = ../libpurple.la $(GLIB_LIBS)
 
 if ENABLE_DBUS
 
 CLEANFILES              = dbus-example-bindings.c
 dbus_example_la_SOURCES = dbus-example.c
 
-dbus_example_la_LIBADD      = $(GLIB_LIBS) $(DBUS_LIBS)
+dbus_example_la_LIBADD      = ../libpurple.la $(GLIB_LIBS) $(DBUS_LIBS)
 
 .PHONY: always
 
--- origsrc/pidgin-2.5.6/libpurple/plugins/mono/loader/Makefile.am	2009-04-28 13:09:24.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/plugins/mono/loader/Makefile.am	2009-06-15 13:46:06.502014900 -0500
@@ -12,9 +12,9 @@ mono_la_SOURCES	= \
 	blist-glue.c \
 	status-glue.c 
 
-mono_la_LDFLAGS  = -module -avoid-version
+mono_la_LDFLAGS  = -module -avoid-version -no-undefined
 
-mono_la_LIBADD = $(MONO_LIBS)
+mono_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(MONO_LIBS)
 
 AM_CPPFLAGS = \
 	-I$(top_srcdir) \
--- origsrc/pidgin-2.5.6/libpurple/plugins/perl/Makefile.am	2009-04-28 13:09:24.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/plugins/perl/Makefile.am	2009-06-15 13:46:06.505015100 -0500
@@ -4,8 +4,8 @@ perl_dirs = common
 
 plugin_LTLIBRARIES = perl.la
 
-perl_la_LDFLAGS = -module -avoid-version
-perl_la_LIBADD = $(GLIB_LIBS) $(PERL_LIBS)
+perl_la_LDFLAGS = -module -avoid-version -no-undefined
+perl_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(PERL_LIBS)
 perl_la_SOURCES = \
 	perl.c \
 	perl-common.c \
--- origsrc/pidgin-2.5.6/libpurple/plugins/perl/common/Makefile.PL.in	2009-04-28 13:09:24.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/plugins/perl/common/Makefile.PL.in	2009-06-15 13:46:06.507015200 -0500
@@ -11,6 +11,7 @@ WriteMakefile(
        AUTHOR           => 'Purple <http://pidgin.im/>') : ()),
     'DEFINE'            => '@DEBUG_CFLAGS@',
     'INC'               => '-I. -I@srcdir@ -I@top_srcdir@ -I@top_srcdir@/libpurple @GLIB_CFLAGS@',
+    'LIBS'              => [ '-L@top_builddir@/libpurple/.libs -lpurple @GLIB_LIBS@' ],
     'OBJECT'            => '$(O_FILES)', # link all the C files too
 #    'OPTIMIZE'          => '-g', # For debugging
 );
--- origsrc/pidgin-2.5.6/libpurple/plugins/ssl/Makefile.am	2009-04-28 13:09:24.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/plugins/ssl/Makefile.am	2009-06-15 13:46:06.510015400 -0500
@@ -3,9 +3,9 @@ EXTRA_DIST = \
 
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-ssl_la_LDFLAGS        = -module -avoid-version
-ssl_gnutls_la_LDFLAGS = -module -avoid-version
-ssl_nss_la_LDFLAGS    = -module -avoid-version
+ssl_la_LDFLAGS        = -module -avoid-version -no-undefined
+ssl_gnutls_la_LDFLAGS = -module -avoid-version -no-undefined
+ssl_nss_la_LDFLAGS    = -module -avoid-version -no-undefined
 
 if PLUGINS
 
@@ -39,9 +39,9 @@ ssl_la_SOURCES        = ssl.c
 ssl_gnutls_la_SOURCES = ssl-gnutls.c
 ssl_nss_la_SOURCES    = ssl-nss.c
 
-ssl_la_LIBADD        = $(GLIB_LIBS)
-ssl_gnutls_la_LIBADD = $(GLIB_LIBS) $(GNUTLS_LIBS)
-ssl_nss_la_LIBADD    = $(GLIB_LIBS) $(NSS_LIBS)
+ssl_la_LIBADD        = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
+ssl_gnutls_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(GNUTLS_LIBS)
+ssl_nss_la_LIBADD    = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(NSS_LIBS)
 
 endif # PLUGINS
 
--- origsrc/pidgin-2.5.6/libpurple/plugins/tcl/Makefile.am	2009-04-28 13:09:24.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/plugins/tcl/Makefile.am	2009-06-15 13:46:06.513015600 -0500
@@ -1,13 +1,13 @@
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-tcl_la_LDFLAGS = -module -avoid-version
+tcl_la_LDFLAGS = -module -avoid-version -no-undefined
 
 plugin_LTLIBRARIES = tcl.la
 
 tcl_la_SOURCES = tcl.c tcl_glib.c tcl_glib.h tcl_cmds.c tcl_signals.c tcl_purple.h \
                  tcl_ref.c tcl_cmd.c
 
-tcl_la_LIBADD = $(GLIB_LIBS) $(TCL_LIBS) $(TK_LIBS)
+tcl_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(TCL_LIBS) $(TK_LIBS)
 
 EXTRA_DIST = signal-test.tcl Makefile.mingw
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/bonjour/Makefile.am	2009-04-28 13:09:25.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/bonjour/Makefile.am	2009-06-15 13:46:06.515015700 -0500
@@ -25,7 +25,7 @@ BONJOURSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libbonjour_la_LDFLAGS = -module -avoid-version
+libbonjour_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_BONJOUR
 
@@ -40,7 +40,7 @@ else
 st =
 pkg_LTLIBRARIES       = libbonjour.la
 libbonjour_la_SOURCES = $(BONJOURSOURCES)
-libbonjour_la_LIBADD  = $(GLIB_LIBS) $(LIBXML_LIBS) $(AVAHI_LIBS)
+libbonjour_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(LIBXML_LIBS) $(AVAHI_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/gg/Makefile.am	2009-04-28 13:09:25.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/gg/Makefile.am	2009-06-15 13:46:06.534016800 -0500
@@ -47,7 +47,7 @@ GGSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libgg_la_LDFLAGS = -module -avoid-version
+libgg_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_GG
 
@@ -62,7 +62,7 @@ else
 st = $(GADU_CFLAGS)
 pkg_LTLIBRARIES = libgg.la
 libgg_la_SOURCES = $(GGSOURCES)
-libgg_la_LIBADD  = $(GLIB_LIBS) $(GADU_LIBS)
+libgg_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(GADU_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/irc/Makefile.am	2009-04-28 13:09:25.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/irc/Makefile.am	2009-06-15 13:46:06.537016900 -0500
@@ -13,7 +13,7 @@ IRCSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libirc_la_LDFLAGS = -module -avoid-version
+libirc_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_IRC
 
@@ -27,7 +27,7 @@ else
 st =
 pkg_LTLIBRARIES   = libirc.la
 libirc_la_SOURCES = $(IRCSOURCES)
-libirc_la_LIBADD  = $(GLIB_LIBS)
+libirc_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/jabber/Makefile.am	2009-04-28 13:09:25.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/jabber/Makefile.am	2009-06-15 13:48:16.969477200 -0500
@@ -54,7 +54,7 @@ JABBERSOURCES = auth.c \
 
 AM_CFLAGS = $(st)
 
-libxmpp_la_LDFLAGS = -module -avoid-version
+libxmpp_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_JABBER
 
@@ -68,7 +69,7 @@ else
 st =
-pkg_LTLIBRARIES      = libjabber.la libxmpp.la
-noinst_LTLIBRARIES      = libjabber.la
+pkg_LTLIBRARIES      = libxmpp.la
 libjabber_la_SOURCES = $(JABBERSOURCES)
-libjabber_la_LIBADD  = $(GLIB_LIBS) $(SASL_LIBS) $(LIBXML_LIBS)
+libjabber_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(SASL_LIBS) $(LIBXML_LIBS)
 
 libxmpp_la_SOURCES = libxmpp.c
 libxmpp_la_LIBADD = libjabber.la
--- origsrc/pidgin-2.5.6/libpurple/protocols/msn/Makefile.am	2009-04-28 13:09:25.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/msn/Makefile.am	2009-06-15 13:46:06.544017300 -0500
@@ -69,7 +69,7 @@ MSNSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libmsn_la_LDFLAGS = -module -avoid-version
+libmsn_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_MSN
 
@@ -83,7 +83,7 @@ else
 st =
 pkg_LTLIBRARIES   = libmsn.la
 libmsn_la_SOURCES = $(MSNSOURCES)
-libmsn_la_LIBADD  = $(GLIB_LIBS)
+libmsn_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/msnp9/Makefile.am	2009-04-28 13:09:26.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/msnp9/Makefile.am	2009-06-15 13:46:06.546017500 -0500
@@ -65,7 +65,7 @@ MSNP9SOURCES = \
 
 AM_CFLAGS = $(st)
 
-libmsn_la_LDFLAGS = -module -avoid-version
+libmsn_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_MSN
 
@@ -79,7 +79,7 @@ else
 st =
 pkg_LTLIBRARIES   = libmsn.la
 libmsn_la_SOURCES = $(MSNP9SOURCES)
-libmsn_la_LIBADD  = $(GLIB_LIBS)
+libmsn_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/myspace/Makefile.am	2009-04-28 13:09:26.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/myspace/Makefile.am	2009-06-15 13:46:06.550017700 -0500
@@ -19,7 +19,7 @@ MSIMSOURCES = markup.c \
 
 AM_CFLAGS = $(st)
 
-libmyspace_la_LDFLAGS = -module -avoid-version
+libmyspace_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_MYSPACE
 
@@ -33,7 +33,7 @@ else
 st =
 pkg_LTLIBRARIES       = libmyspace.la
 libmyspace_la_SOURCES = $(MSIMSOURCES)
-libmyspace_la_LIBADD  = $(GLIB_LIBS)
+libmyspace_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/novell/Makefile.am	2009-04-28 13:09:26.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/novell/Makefile.am	2009-06-15 13:46:06.553017900 -0500
@@ -28,7 +28,7 @@ NOVELLSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libnovell_la_LDFLAGS = -module -avoid-version
+libnovell_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_NOVELL
 
@@ -42,7 +42,7 @@ else
 st =
 pkg_LTLIBRARIES      = libnovell.la
 libnovell_la_SOURCES = $(NOVELLSOURCES)
-libnovell_la_LIBADD  = $(GLIB_LIBS)
+libnovell_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/null/Makefile.am	2009-04-28 13:09:26.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/null/Makefile.am	2009-06-15 13:48:43.394988700 -0500
@@ -8,7 +8,7 @@ NULLSOURCES = nullprpl.c
 
 AM_CFLAGS = $(st)
 
-libnull_la_LDFLAGS = -module -avoid-version
+libnull_la_LDFLAGS = -module -avoid-version -no-undefined
 
 # nullprpl isn't built by default. when it is built, it's dynamically linked.
 st =
--- origsrc/pidgin-2.5.6/libpurple/protocols/oscar/Makefile.am	2009-04-28 13:09:26.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/oscar/Makefile.am	2009-06-15 13:46:06.562018400 -0500
@@ -47,8 +47,8 @@ OSCARSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libaim_la_LDFLAGS = -module -avoid-version
-libicq_la_LDFLAGS = -module -avoid-version
+libaim_la_LDFLAGS = -module -avoid-version -no-undefined
+libicq_la_LDFLAGS = -module -avoid-version -no-undefined
 if STATIC_OSCAR
 
 st = -DPURPLE_STATIC_PRPL
@@ -61,7 +62,7 @@ liboscar_la_CFLAGS  = $(AM_CFLAGS)
 st =
-pkg_LTLIBRARIES     = liboscar.la libaim.la libicq.la
+noinst_LTLIBRARIES  = liboscar.la
+pkg_LTLIBRARIES     = libaim.la libicq.la
 liboscar_la_SOURCES = $(OSCARSOURCES)
-liboscar_la_LIBADD  = $(GLIB_LIBS)
+liboscar_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 libaim_la_SOURCES   = libaim.c
 libaim_la_LIBADD    = liboscar.la
--- origsrc/pidgin-2.5.6/libpurple/protocols/qq/Makefile.am	2009-04-28 13:09:26.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/qq/Makefile.am	2009-06-15 13:46:06.566018600 -0500
@@ -53,7 +53,7 @@ QQSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libqq_la_LDFLAGS = -module -avoid-version
+libqq_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_QQ
 
@@ -67,7 +67,7 @@ else
 st =
 pkg_LTLIBRARIES  = libqq.la
 libqq_la_SOURCES = $(QQSOURCES)
-libqq_la_LIBADD  = $(GLIB_LIBS)
+libqq_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/sametime/Makefile.am	2009-04-28 13:09:27.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/sametime/Makefile.am	2009-06-15 13:49:04.179177500 -0500
@@ -24,8 +24,8 @@ pkg_LTLIBRARIES     = libsametime.la
 endif
 
 libsametime_la_SOURCES = $(SAMETIMESOURCES)
-libsametime_la_LDFLAGS = -module -avoid-version
-libsametime_la_LIBADD = $(GLIB_LIBS) $(MEANWHILE_LIBS)
+libsametime_la_LDFLAGS = -module -avoid-version -no-undefined
+libsametime_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(MEANWHILE_LIBS)
 
 AM_CPPFLAGS = \
 	-I$(top_srcdir)/libpurple \
--- origsrc/pidgin-2.5.6/libpurple/protocols/silc/Makefile.am	2009-04-28 13:09:27.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/silc/Makefile.am	2009-06-15 13:46:06.575019100 -0500
@@ -19,7 +19,7 @@ SILCSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libsilcpurple_la_LDFLAGS = -module -avoid-version
+libsilcpurple_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_SILC
 
@@ -34,7 +34,7 @@ else
 st = $(SILC_CFLAGS)
 pkg_LTLIBRARIES          = libsilcpurple.la
 libsilcpurple_la_SOURCES = $(SILCSOURCES)
-libsilcpurple_la_LIBADD  = $(GLIB_LIBS) $(SILC_LIBS)
+libsilcpurple_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(SILC_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/silc10/Makefile.am	2009-04-28 13:09:27.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/silc10/Makefile.am	2009-06-15 13:46:06.578019300 -0500
@@ -9,7 +9,7 @@ SILCSOURCES = silc.c silcpurple.h buddy.
 
 AM_CFLAGS = $(st)
 
-libsilcpurple_la_LDFLAGS = -module -avoid-version
+libsilcpurple_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_SILC
 
@@ -24,7 +24,7 @@ else
 st = $(SILC_CFLAGS)
 pkg_LTLIBRARIES          = libsilcpurple.la
 libsilcpurple_la_SOURCES = $(SILCSOURCES)
-libsilcpurple_la_LIBADD  = $(GLIB_LIBS) $(SILC_LIBS)
+libsilcpurple_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(SILC_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/simple/Makefile.am	2009-04-28 13:09:27.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/simple/Makefile.am	2009-06-15 13:46:06.581019500 -0500
@@ -11,7 +11,7 @@ SIMPLESOURCES = \
 
 AM_CFLAGS = $(st)
 
-libsimple_la_LDFLAGS = -module -avoid-version
+libsimple_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_MSN
 
@@ -25,7 +25,7 @@ else
 st =
 pkg_LTLIBRARIES      = libsimple.la
 libsimple_la_SOURCES = $(SIMPLESOURCES)
-libsimple_la_LIBADD  = $(GLIB_LIBS)
+libsimple_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/toc/Makefile.am	2009-04-28 13:09:27.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/toc/Makefile.am	2009-06-15 13:46:06.583019600 -0500
@@ -8,7 +8,7 @@ TOCSOURCES = toc.c
 
 AM_CFLAGS = $(st)
 
-libtoc_la_LDFLAGS = -module -avoid-version
+libtoc_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_TOC
 
@@ -22,6 +22,7 @@ else
 st =
 pkg_LTLIBRARIES   = libtoc.la
 libtoc_la_SOURCES = $(TOCSOURCES)
+libtoc_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/yahoo/Makefile.am	2009-04-28 13:09:27.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/yahoo/Makefile.am	2009-06-15 13:46:06.586019700 -0500
@@ -31,7 +31,7 @@ YAHOOSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libyahoo_la_LDFLAGS = -module -avoid-version
+libyahoo_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_YAHOO
 
@@ -45,7 +45,7 @@ else
 st =
 pkg_LTLIBRARIES     = libyahoo.la
 libyahoo_la_SOURCES = $(YAHOOSOURCES)
-libyahoo_la_LIBADD  = $(GLIB_LIBS)
+libyahoo_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/libpurple/protocols/zephyr/Makefile.am	2009-04-28 13:09:27.000000000 -0500
+++ src/pidgin-2.5.6/libpurple/protocols/zephyr/Makefile.am	2009-06-15 13:46:06.589019900 -0500
@@ -72,7 +72,7 @@ ZEPHYRLIBS = $(KRB4_LDFLAGS) $(KRB4_LIBS
 
 ZEPHYRLIBSEXT = $(ZEPHYR_LDFLAGS) $(ZEPHYR_LIBS)
 
-libzephyr_la_LDFLAGS = -module -avoid-version
+libzephyr_la_LDFLAGS = -module -avoid-version -no-undefined
 
 if STATIC_ZEPHYR
 
@@ -89,10 +89,10 @@ pkg_LTLIBRARIES = libzephyr.la
 
 if EXTERNAL_LIBZEPHYR
 libzephyr_la_SOURCES = $(ZEPHYRSOURCESEXT)
-libzephyr_la_LIBADD  = $(GLIB_LIBS) $(ZEPHYRLIBSEXT)
+libzephyr_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(ZEPHYRLIBSEXT)
 else
 libzephyr_la_SOURCES = $(ZEPHYRSOURCES)
-libzephyr_la_LIBADD  = $(GLIB_LIBS) $(ZEPHYRLIBS)
+libzephyr_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS) $(ZEPHYRLIBS)
 endif
 
 
--- origsrc/pidgin-2.5.6/pidgin/Makefile.am	2009-04-28 13:09:27.000000000 -0500
+++ src/pidgin-2.5.6/pidgin/Makefile.am	2009-06-15 13:46:06.595020300 -0500
@@ -68,7 +68,7 @@ if ENABLE_GTK
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = pidgin.pc
 
-SUBDIRS = pixmaps plugins
+SUBDIRS = . pixmaps plugins
 
 bin_PROGRAMS = pidgin
 
@@ -184,7 +184,13 @@ pidgininclude_HEADERS = \
 
 
 pidgin_DEPENDENCIES = @LIBOBJS@
+
+if PLATFORM_WIN32
+pidgin_LDFLAGS = -Wl,--export-all-symbols,--out-implib,libpidgin.a
+else
 pidgin_LDFLAGS = -export-dynamic
+endif
+
 pidgin_LDADD = \
 	@LIBOBJS@ \
 	$(GLIB_LIBS) \
@@ -223,3 +229,9 @@ AM_CPPFLAGS = \
 	$(LIBXML_CFLAGS) \
 	$(INTGG_CFLAGS)
 endif  # ENABLE_GTK
+
+if PLATFORM_WIN32
+install-data-local:
+	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) libpidgin.a $(DESTDIR)$(libdir)
+endif
--- origsrc/pidgin-2.5.6/pidgin/pidgin-uninstalled.pc.in	2009-04-28 13:09:28.000000000 -0500
+++ src/pidgin-2.5.6/pidgin/pidgin-uninstalled.pc.in	2009-06-15 13:46:06.597020400 -0500
@@ -11,3 +11,4 @@ Description: Pidgin is a GTK2-based inst
 Version: @VERSION@
 Requires: gtk+-2.0 purple
 Cflags: -I${pcfiledir}
+Libs: -L${pcfiledir} @LIBPIDGIN@
--- origsrc/pidgin-2.5.6/pidgin/pidgin.pc.in	2009-04-28 13:09:28.000000000 -0500
+++ src/pidgin-2.5.6/pidgin/pidgin.pc.in	2009-06-15 13:46:06.599020500 -0500
@@ -11,4 +11,5 @@ Description: Pidgin is a GTK2-based inst
 Version: @VERSION@
 Requires: gtk+-2.0 purple
 Cflags: -I${includedir}/pidgin
+Libs: -L${libdir} @LIBPIDGIN@
 
--- origsrc/pidgin-2.5.6/pidgin/plugins/Makefile.am	2009-04-28 13:09:29.000000000 -0500
+++ src/pidgin-2.5.6/pidgin/plugins/Makefile.am	2009-06-15 13:46:06.603020700 -0500
@@ -30,22 +30,22 @@ SUBDIRS = \
 
 plugindir = $(libdir)/pidgin
 
-convcolors_la_LDFLAGS       = -module -avoid-version
-contact_priority_la_LDFLAGS = -module -avoid-version
-extplacement_la_LDFLAGS     = -module -avoid-version
-gtk_signals_test_la_LDFLAGS = -module -avoid-version
-gtkbuddynote_la_LDFLAGS     = -module -avoid-version
-history_la_LDFLAGS          = -module -avoid-version
-iconaway_la_LDFLAGS         = -module -avoid-version
-markerline_la_LDFLAGS       = -module -avoid-version
-notify_la_LDFLAGS           = -module -avoid-version
-pidginrc_la_LDFLAGS         = -module -avoid-version
-relnot_la_LDFLAGS           = -module -avoid-version
-sendbutton_la_LDFLAGS       = -module -avoid-version
-spellchk_la_LDFLAGS         = -module -avoid-version
-timestamp_la_LDFLAGS        = -module -avoid-version
-timestamp_format_la_LDFLAGS = -module -avoid-version
-xmppconsole_la_LDFLAGS      = -module -avoid-version
+convcolors_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+contact_priority_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+extplacement_la_LDFLAGS     = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+gtk_signals_test_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+gtkbuddynote_la_LDFLAGS     = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+history_la_LDFLAGS          = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+iconaway_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+markerline_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+notify_la_LDFLAGS           = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+pidginrc_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+relnot_la_LDFLAGS           = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+sendbutton_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+spellchk_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+timestamp_la_LDFLAGS        = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+timestamp_format_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+xmppconsole_la_LDFLAGS      = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -86,22 +86,22 @@ timestamp_la_SOURCES        = timestamp.
 timestamp_format_la_SOURCES = timestamp_format.c
 xmppconsole_la_SOURCES      = xmppconsole.c
 
-convcolors_la_LIBADD        = $(GTK_LIBS)
-contact_priority_la_LIBADD  = $(GTK_LIBS)
-extplacement_la_LIBADD      = $(GTK_LIBS)
-gtk_signals_test_la_LIBADD  = $(GTK_LIBS)
-gtkbuddynote_la_LIBADD      = $(GTK_LIBS)
-history_la_LIBADD           = $(GTK_LIBS)
-iconaway_la_LIBADD          = $(GTK_LIBS)
-markerline_la_LIBADD        = $(GTK_LIBS)
-notify_la_LIBADD            = $(GTK_LIBS)
-pidginrc_la_LIBADD          = $(GTK_LIBS)
-relnot_la_LIBADD            = $(GLIB_LIBS)
-sendbutton_la_LIBADD        = $(GTK_LIBS)
-spellchk_la_LIBADD          = $(GTK_LIBS)
-timestamp_la_LIBADD         = $(GTK_LIBS)
-timestamp_format_la_LIBADD  = $(GTK_LIBS)
-xmppconsole_la_LIBADD       = $(GTK_LIBS)
+convcolors_la_LIBADD        = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+contact_priority_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+extplacement_la_LIBADD      = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+gtk_signals_test_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+gtkbuddynote_la_LIBADD      = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+history_la_LIBADD           = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+iconaway_la_LIBADD          = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+markerline_la_LIBADD        = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+notify_la_LIBADD            = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+pidginrc_la_LIBADD          = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+relnot_la_LIBADD            = $(top_builddir)/libpurple/libpurple.la $(GLIB_LIBS)
+sendbutton_la_LIBADD        = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+spellchk_la_LIBADD          = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+timestamp_la_LIBADD         = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+timestamp_format_la_LIBADD  = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
+xmppconsole_la_LIBADD       = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
 
 endif # PLUGINS
 
--- origsrc/pidgin-2.5.6/pidgin/plugins/cap/Makefile.am	2009-04-28 13:09:29.000000000 -0500
+++ src/pidgin-2.5.6/pidgin/plugins/cap/Makefile.am	2009-06-15 13:46:06.605020800 -0500
@@ -1,6 +1,6 @@
 capdir = $(libdir)/pidgin
 
-cap_la_LDFLAGS = -module -avoid-version
+cap_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -15,7 +15,7 @@ cap_la_SOURCES = \
 
 endif
 
-cap_la_LIBADD = $(GTK_LIBS) $(SQLITE3_LIBS)
+cap_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS) $(SQLITE3_LIBS)
 
 AM_CPPFLAGS = \
 	-DDATADIR=\"$(datadir)\" \
--- origsrc/pidgin-2.5.6/pidgin/plugins/gestures/Makefile.am	2009-04-28 13:09:30.000000000 -0500
+++ src/pidgin-2.5.6/pidgin/plugins/gestures/Makefile.am	2009-06-15 13:46:06.607020900 -0500
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/pidgin
 
-gestures_la_LDFLAGS = -module -avoid-version
+gestures_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -13,7 +13,7 @@ gestures_la_SOURCES = \
 	stroke.c \
 	stroke-draw.c
 
-gestures_la_LIBADD = $(GTK_LIBS)
+gestures_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/pidgin/plugins/gevolution/Makefile.am	2009-04-28 13:09:30.000000000 -0500
+++ src/pidgin-2.5.6/pidgin/plugins/gevolution/Makefile.am	2009-06-15 13:46:06.609021100 -0500
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/pidgin
 
-gevolution_la_LDFLAGS = -module -avoid-version
+gevolution_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -15,7 +15,7 @@ gevolution_la_SOURCES = \
 	new_person_dialog.c \
 	eds-utils.c
 
-gevolution_la_LIBADD = $(EVOLUTION_ADDRESSBOOK_LIBS) $(GTK_LIBS)
+gevolution_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(EVOLUTION_ADDRESSBOOK_LIBS) $(GTK_LIBS)
 
 endif
 
--- origsrc/pidgin-2.5.6/pidgin/plugins/musicmessaging/Makefile.am	2009-04-28 13:09:30.000000000 -0500
+++ src/pidgin-2.5.6/pidgin/plugins/musicmessaging/Makefile.am	2009-06-15 13:46:06.612021200 -0500
@@ -3,7 +3,7 @@ EXTRA_DIST = \
 
 musicmessagingdir = $(libdir)/pidgin
 
-musicmessaging_la_LDFLAGS = -module -avoid-version
+musicmessaging_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 if ENABLE_DBUS
@@ -16,7 +16,7 @@ musicmessaging_LTLIBRARIES = musicmessag
 musicmessaging_la_SOURCES = \
 	musicmessaging.c
 
-musicmessaging_la_LIBADD = $(GTK_LIBS) $(DBUS_LIBS)
+musicmessaging_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS) $(DBUS_LIBS)
 
 CLEANFILES              = music-messaging-bindings.c
 
--- origsrc/pidgin-2.5.6/pidgin/plugins/perl/common/Makefile.PL.in	2009-04-28 13:09:30.000000000 -0500
+++ src/pidgin-2.5.6/pidgin/plugins/perl/common/Makefile.PL.in	2009-06-15 13:46:06.614021300 -0500
@@ -10,6 +10,7 @@ WriteMakefile(
        'AUTHOR'        => 'Pidgin <http://pidgin.im/>') :  ()),
     'DEFINE'        => '@DEBUG_CFLAGS@',
     'INC'           => '-I. -I@srcdir@ -I@top_srcdir@ -I@top_srcdir@/libpurple -I@top_srcdir@/pidgin @GTK_CFLAGS@',
+    'LIBS'          => ['-L@top_builddir@/libpurple/.libs -lpurple -L@top_builddir@/pidgin @LIBPIDGIN@ @GLIB_LIBS@'],
     'OBJECT'        => '$(O_FILES)', # link all the C files too
     'TYPEMAPS'      => ["@top_srcdir@/libpurple/plugins/perl/common/typemap"],
 #    'OPTIMIZE'      => '-g', # For debugging.
--- origsrc/pidgin-2.5.6/pidgin/plugins/ticker/Makefile.am	2009-04-28 13:09:30.000000000 -0500
+++ src/pidgin-2.5.6/pidgin/plugins/ticker/Makefile.am	2009-06-15 13:46:06.616021500 -0500
@@ -3,7 +3,7 @@ EXTRA_DIST = \
 
 plugindir = $(libdir)/pidgin
 
-ticker_la_LDFLAGS = -module -avoid-version
+ticker_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
@@ -14,7 +14,7 @@ ticker_la_SOURCES = \
 	gtkticker.h \
 	ticker.c
 
-ticker_la_LIBADD = $(GTK_LIBS)
+ticker_la_LIBADD = $(top_builddir)/libpurple/libpurple.la $(GTK_LIBS)
 
 endif
 
