--- origsrc/pidgin-2.6.4/configure.ac	2009-11-29 21:13:44.000000000 -0600
+++ src/pidgin-2.6.4/configure.ac	2009-12-19 22:14:16.956896700 -0600
@@ -118,6 +118,33 @@ AC_FUNC_ALLOCA
 GETTEXT_PACKAGE=pidgin
 AC_SUBST(GETTEXT_PACKAGE)
 
+# win32
+case "$host_os" in
+cygwin*)
+	platform_win32=yes
+	LIBFINCH="-lfinch"
+	LIBPIDGIN="-lpidgin"
+	FINCH_PLUGIN_LDFLAGS="-no-undefined -Wl,\$(top_builddir)/finch/libfinch.a \$(top_builddir)/finch/libgnt/libgnt.la \$(top_builddir)/libpurple/libpurple.la"
+	PIDGIN_PLUGIN_LDFLAGS="-no-undefined -Wl,\$(top_builddir)/pidgin/libpidgin.a \$(top_builddir)/libpurple/libpurple.la"
+	PURPLE_PLUGIN_LDFLAGS="-no-undefined \$(top_builddir)/libpurple/libpurple.la"
+	;;
+*)
+	platform_win32=no
+	LIBFINCH=
+	LIBPIDGIN=
+	LIBPIDGIN=
+	FINCH_PLUGIN_LDFLAGS=
+	PIDGIN_PLUGIN_LDFLAGS=
+	PURPLE_PLUGIN_LDFLAGS=
+	;;
+esac
+
+AM_CONDITIONAL(PLATFORM_WIN32, test "x$platform_win32" = "xyes")
+AC_SUBST(LIBFINCH)
+AC_SUBST(LIBPIDGIN)
+AC_SUBST(FINCH_PLUGIN_LDFLAGS)
+AC_SUBST(PIDGIN_PLUGIN_LDFLAGS)
+AC_SUBST(PURPLE_PLUGIN_LDFLAGS)
 
 # before gettexting, in case iconv matters
 case "$host_os" in
--- origsrc/pidgin-2.6.4/finch/Makefile.am	2009-11-29 21:13:44.000000000 -0600
+++ src/pidgin-2.6.4/finch/Makefile.am	2009-12-19 22:14:16.963897100 -0600
@@ -7,7 +7,7 @@ EXTRA_DIST = \
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = finch.pc
 
-SUBDIRS = libgnt plugins
+SUBDIRS = libgnt . plugins
 
 if ENABLE_GNT
 
@@ -64,7 +64,13 @@ finchinclude_HEADERS = \
 	$(finch_headers)
 
 finch_DEPENDENCIES = @LIBOBJS@
+
+if PLATFORM_WIN32
+finch_LDFLAGS = -Wl,--export-all-symbols -Wl,--out-implib,libfinch.a
+else
 finch_LDFLAGS = -export-dynamic
+endif
+
 finch_LDADD = \
 	@LIBOBJS@ \
 	$(DBUS_LIBS) \
@@ -91,3 +97,9 @@ AM_CPPFLAGS = \
 	$(LIBXML_CFLAGS) \
 	$(GSTREAMER_CFLAGS) \
 	$(GNT_CFLAGS)
+
+if PLATFORM_WIN32
+install-data-local:
+	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) libfinch.a $(DESTDIR)$(libdir)
+endif
--- origsrc/pidgin-2.6.4/finch/finch.pc.in	2009-11-29 21:13:45.000000000 -0600
+++ src/pidgin-2.6.4/finch/finch.pc.in	2009-12-19 22:14:16.967897300 -0600
@@ -11,4 +11,5 @@ Description: Finch is an instant messeng
 Version: @VERSION@
 Requires: gnt purple
 Cflags: -I${includedir}/finch
+Libs: -L${libdir} @LIBFINCH@
 
--- origsrc/pidgin-2.6.4/finch/libgnt/wms/Makefile.am	2009-11-29 21:13:46.000000000 -0600
+++ src/pidgin-2.6.4/finch/libgnt/wms/Makefile.am	2009-12-19 22:14:16.971897500 -0600
@@ -6,8 +6,8 @@ else
 purple_wms =
 endif
 
-s_la_LDFLAGS             = -module -avoid-version
-irssi_la_LDFLAGS         = -module -avoid-version
+s_la_LDFLAGS             = -module -avoid-version -no-undefined
+irssi_la_LDFLAGS         = -module -avoid-version -no-undefined
 
 plugin_LTLIBRARIES = \
 	$(wms) \
--- origsrc/pidgin-2.6.4/finch/plugins/Makefile.am	2009-11-29 21:13:46.000000000 -0600
+++ src/pidgin-2.6.4/finch/plugins/Makefile.am	2009-12-19 22:14:16.976897800 -0600
@@ -1,9 +1,9 @@
-gntclipboard_la_LDFLAGS = -module -avoid-version
-gntgf_la_LDFLAGS      = -module -avoid-version
-gnthistory_la_LDFLAGS = -module -avoid-version
-gntlastlog_la_LDFLAGS = -module -avoid-version
-gnttinyurl_la_LDFLAGS = -module -avoid-version
-grouping_la_LDFLAGS   = -module -avoid-version
+gntclipboard_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gntgf_la_LDFLAGS      = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gnthistory_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gntlastlog_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+gnttinyurl_la_LDFLAGS = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
+grouping_la_LDFLAGS   = -module -avoid-version $(FINCH_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.6.4/libpurple/Makefile.am	2009-11-29 21:13:46.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/Makefile.am	2009-12-19 22:13:55.490668900 -0600
@@ -32,7 +32,7 @@ endif
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = purple.pc
 
-SUBDIRS = $(GCONF_DIR) plugins protocols . tests example
+SUBDIRS = . $(GCONF_DIR) plugins protocols tests example
 
 purple_coresources = \
 	account.c \
--- origsrc/pidgin-2.6.4/libpurple/plugins/Makefile.am	2009-11-29 21:13:47.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/plugins/Makefile.am	2009-12-19 22:14:16.981898100 -0600
@@ -24,27 +24,27 @@ SUBDIRS = \
 
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-autoaccept_la_LDFLAGS       = -module -avoid-version
-buddynote_la_LDFLAGS        = -module -avoid-version
-ciphertest_la_LDFLAGS		= -module -avoid-version
-codeinline_la_LDFLAGS		= -module -avoid-version
-debug_example_la_LDFLAGS    = -module -avoid-version
-helloworld_la_LDFLAGS       = -module -avoid-version
-idle_la_LDFLAGS             = -module -avoid-version
-joinpart_la_LDFLAGS         = -module -avoid-version
-log_reader_la_LDFLAGS       = -module -avoid-version
-newline_la_LDFLAGS          = -module -avoid-version
-notify_example_la_LDFLAGS   = -module -avoid-version
-offlinemsg_la_LDFLAGS       = -module -avoid-version
-one_time_password_la_LDFLAGS	= -module -avoid-version
-pluginpref_example_la_LDFLAGS = -module -avoid-version
-psychic_la_LDFLAGS          = -module -avoid-version
-signals_test_la_LDFLAGS		= -module -avoid-version
-simple_la_LDFLAGS			= -module -avoid-version
-statenotify_la_LDFLAGS      = -module -avoid-version
+autoaccept_la_LDFLAGS       = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+buddynote_la_LDFLAGS        = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+ciphertest_la_LDFLAGS		= -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+codeinline_la_LDFLAGS		= -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+debug_example_la_LDFLAGS    = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+helloworld_la_LDFLAGS       = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+idle_la_LDFLAGS             = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+joinpart_la_LDFLAGS         = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+log_reader_la_LDFLAGS       = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+newline_la_LDFLAGS          = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+notify_example_la_LDFLAGS   = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+offlinemsg_la_LDFLAGS       = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+one_time_password_la_LDFLAGS	= -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+pluginpref_example_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+psychic_la_LDFLAGS          = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+signals_test_la_LDFLAGS		= -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+simple_la_LDFLAGS			= -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+statenotify_la_LDFLAGS      = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 # this can't be in a conditional otherwise automake 1.4 yells
-dbus_example_la_LDFLAGS     = -module -avoid-version
+dbus_example_la_LDFLAGS     = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.6.4/libpurple/plugins/mono/loader/Makefile.am	2009-11-29 21:13:48.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/plugins/mono/loader/Makefile.am	2009-12-19 22:14:16.986898400 -0600
@@ -12,7 +12,7 @@ mono_la_SOURCES	= \
 	blist-glue.c \
 	status-glue.c 
 
-mono_la_LDFLAGS  = -module -avoid-version
+mono_la_LDFLAGS  = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 mono_la_LIBADD = $(MONO_LIBS)
 
--- origsrc/pidgin-2.6.4/libpurple/plugins/perl/Makefile.am	2009-11-29 21:13:48.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/plugins/perl/Makefile.am	2009-12-19 22:14:16.990898600 -0600
@@ -4,7 +4,7 @@ perl_dirs = common
 
 plugin_LTLIBRARIES = perl.la
 
-perl_la_LDFLAGS = -module -avoid-version
+perl_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 perl_la_LIBADD = $(GLIB_LIBS) $(PERL_LIBS)
 perl_la_SOURCES = \
 	perl.c \
--- origsrc/pidgin-2.6.4/libpurple/plugins/perl/common/Makefile.PL.in	2009-11-29 21:13:48.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/plugins/perl/common/Makefile.PL.in	2009-12-19 22:14:16.995898900 -0600
@@ -11,6 +11,7 @@ WriteMakefile(
        AUTHOR           => 'Purple <http://pidgin.im/>') : ()),
     'DEFINE'            => '@DEBUG_CFLAGS@',
     'INC'               => '-I. -I@srcdir@ -I@top_srcdir@ -I@top_srcdir@/libpurple @GLIB_CFLAGS@',
+    'LIBS'              => [ '-L@top_builddir@/libpurple/.libs -lpurple @GLIB_LIBS@' ],
     'OBJECT'            => '$(O_FILES)', # link all the C files too
 #    'OPTIMIZE'          => '-g', # For debugging
     'INSTALLDIRS'          => 'vendor',
--- origsrc/pidgin-2.6.4/libpurple/plugins/ssl/Makefile.am	2009-11-29 21:13:48.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/plugins/ssl/Makefile.am	2009-12-19 22:14:16.999899100 -0600
@@ -3,9 +3,9 @@ EXTRA_DIST = \
 
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-ssl_la_LDFLAGS        = -module -avoid-version
-ssl_gnutls_la_LDFLAGS = -module -avoid-version
-ssl_nss_la_LDFLAGS    = -module -avoid-version
+ssl_la_LDFLAGS        = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+ssl_gnutls_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+ssl_nss_la_LDFLAGS    = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.6.4/libpurple/plugins/tcl/Makefile.am	2009-11-29 21:13:49.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/plugins/tcl/Makefile.am	2009-12-19 22:14:17.004899400 -0600
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/purple-$(PURPLE_MAJOR_VERSION)
 
-tcl_la_LDFLAGS = -module -avoid-version
+tcl_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 plugin_LTLIBRARIES = tcl.la
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/bonjour/Makefile.am	2009-11-29 21:13:49.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/bonjour/Makefile.am	2009-12-19 22:14:17.009899700 -0600
@@ -25,7 +25,7 @@ BONJOURSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libbonjour_la_LDFLAGS = -module -avoid-version
+libbonjour_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_BONJOUR
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/gg/Makefile.am	2009-11-29 21:13:49.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/gg/Makefile.am	2009-12-19 22:14:17.014900000 -0600
@@ -47,7 +47,7 @@ GGSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libgg_la_LDFLAGS = -module -avoid-version
+libgg_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_GG
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/irc/Makefile.am	2009-11-29 21:13:49.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/irc/Makefile.am	2009-12-19 22:14:17.018900200 -0600
@@ -13,7 +13,7 @@ IRCSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libirc_la_LDFLAGS = -module -avoid-version
+libirc_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_IRC
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/jabber/Makefile.am	2009-11-29 21:13:49.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/jabber/Makefile.am	2009-12-19 22:14:17.032901000 -0600
@@ -76,7 +76,7 @@ JABBERSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libxmpp_la_LDFLAGS = -module -avoid-version
+libxmpp_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_JABBER
 
@@ -88,7 +88,8 @@ libjabber_la_CFLAGS  = $(AM_CFLAGS)
 else
 
 st =
-pkg_LTLIBRARIES      = libjabber.la libxmpp.la
+noinst_LTLIBRARIES   = libjabber.la
+pkg_LTLIBRARIES      = libxmpp.la
 libjabber_la_SOURCES = $(JABBERSOURCES)
 libjabber_la_LIBADD  = $(GLIB_LIBS) $(SASL_LIBS) $(LIBXML_LIBS) $(IDN_LIBS)
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/msn/Makefile.am	2009-11-29 21:13:50.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/msn/Makefile.am	2009-12-19 22:14:17.036901200 -0600
@@ -69,7 +69,7 @@ MSNSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libmsn_la_LDFLAGS = -module -avoid-version
+libmsn_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_MSN
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/msnp9/Makefile.am	2009-11-29 21:13:50.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/msnp9/Makefile.am	2009-12-19 22:14:17.041901500 -0600
@@ -65,7 +65,7 @@ MSNP9SOURCES = \
 
 AM_CFLAGS = $(st)
 
-libmsn_la_LDFLAGS = -module -avoid-version
+libmsn_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_MSN
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/mxit/Makefile.am	2009-11-29 21:13:52.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/mxit/Makefile.am	2009-12-19 22:15:25.909840500 -0600
@@ -38,7 +38,7 @@ MXITSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libmxit_la_LDFLAGS = -module -avoid-version
+libmxit_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_MXIT
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/myspace/Makefile.am	2009-11-29 21:13:52.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/myspace/Makefile.am	2009-12-19 22:14:17.045901800 -0600
@@ -19,7 +19,7 @@ MSIMSOURCES = markup.c \
 
 AM_CFLAGS = $(st)
 
-libmyspace_la_LDFLAGS = -module -avoid-version
+libmyspace_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_MYSPACE
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/novell/Makefile.am	2009-11-29 21:13:52.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/novell/Makefile.am	2009-12-19 22:14:17.050902000 -0600
@@ -28,7 +28,7 @@ NOVELLSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libnovell_la_LDFLAGS = -module -avoid-version
+libnovell_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_NOVELL
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/null/Makefile.am	2009-11-29 21:13:52.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/null/Makefile.am	2009-12-19 22:14:17.054902300 -0600
@@ -8,7 +8,7 @@ NULLSOURCES = nullprpl.c
 
 AM_CFLAGS = $(st)
 
-libnull_la_LDFLAGS = -module -avoid-version
+libnull_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 # nullprpl isn't built by default. when it is built, it's dynamically linked.
 st =
--- origsrc/pidgin-2.6.4/libpurple/protocols/oscar/Makefile.am	2009-11-29 21:13:52.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/oscar/Makefile.am	2009-12-19 22:14:17.059902600 -0600
@@ -48,8 +48,8 @@ OSCARSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libaim_la_LDFLAGS = -module -avoid-version
-libicq_la_LDFLAGS = -module -avoid-version
+libaim_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+libicq_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 if STATIC_OSCAR
 
 st = -DPURPLE_STATIC_PRPL
@@ -60,7 +60,8 @@ liboscar_la_CFLAGS  = $(AM_CFLAGS)
 else
 
 st =
-pkg_LTLIBRARIES     = liboscar.la libaim.la libicq.la
+noinst_LTLIBRARIES  = liboscar.la
+pkg_LTLIBRARIES     = libaim.la libicq.la
 liboscar_la_SOURCES = $(OSCARSOURCES)
 liboscar_la_LIBADD  = $(GLIB_LIBS)
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/qq/Makefile.am	2009-11-29 21:13:53.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/qq/Makefile.am	2009-12-19 22:14:17.063902800 -0600
@@ -53,7 +53,7 @@ QQSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libqq_la_LDFLAGS = -module -avoid-version
+libqq_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_QQ
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/sametime/Makefile.am	2009-11-29 21:13:53.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/sametime/Makefile.am	2009-12-19 22:14:17.068903100 -0600
@@ -24,7 +24,7 @@ pkg_LTLIBRARIES     = libsametime.la
 endif
 
 libsametime_la_SOURCES = $(SAMETIMESOURCES)
-libsametime_la_LDFLAGS = -module -avoid-version
+libsametime_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 libsametime_la_LIBADD = $(GLIB_LIBS) $(MEANWHILE_LIBS)
 
 AM_CPPFLAGS = \
--- origsrc/pidgin-2.6.4/libpurple/protocols/silc/Makefile.am	2009-11-29 21:13:53.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/silc/Makefile.am	2009-12-19 22:14:17.072903300 -0600
@@ -19,7 +19,7 @@ SILCSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libsilcpurple_la_LDFLAGS = -module -avoid-version
+libsilcpurple_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_SILC
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/silc10/Makefile.am	2009-11-29 21:13:53.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/silc10/Makefile.am	2009-12-19 22:14:17.077903600 -0600
@@ -9,7 +9,7 @@ SILCSOURCES = silc.c silcpurple.h buddy.
 
 AM_CFLAGS = $(st)
 
-libsilcpurple_la_LDFLAGS = -module -avoid-version
+libsilcpurple_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_SILC
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/simple/Makefile.am	2009-11-29 21:13:53.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/simple/Makefile.am	2009-12-19 22:14:17.081903800 -0600
@@ -11,7 +11,7 @@ SIMPLESOURCES = \
 
 AM_CFLAGS = $(st)
 
-libsimple_la_LDFLAGS = -module -avoid-version
+libsimple_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_SIMPLE
 
--- origsrc/pidgin-2.6.4/libpurple/protocols/yahoo/Makefile.am	2009-11-29 21:13:53.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/yahoo/Makefile.am	2009-12-19 22:14:17.085904000 -0600
@@ -27,8 +27,8 @@ YAHOOSOURCES = \
 
 AM_CFLAGS = $(st)
 
-libyahoo_la_LDFLAGS = -module -avoid-version
-libyahoojp_la_LDFLAGS = -module -avoid-version
+libyahoo_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
+libyahoojp_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_YAHOO
 
@@ -40,7 +40,8 @@ libymsg_la_CFLAGS  = $(AM_CFLAGS)
 else
 
 st =
-pkg_LTLIBRARIES     = libymsg.la libyahoo.la libyahoojp.la
+noinst_LTLIBRARIES  = libymsg.la
+pkg_LTLIBRARIES     = libyahoo.la libyahoojp.la
 
 libymsg_la_SOURCES = $(YAHOOSOURCES)
 libymsg_la_LIBADD  = $(GLIB_LIBS)
--- origsrc/pidgin-2.6.4/libpurple/protocols/zephyr/Makefile.am	2009-11-29 21:13:54.000000000 -0600
+++ src/pidgin-2.6.4/libpurple/protocols/zephyr/Makefile.am	2009-12-19 22:14:17.090904300 -0600
@@ -72,7 +72,7 @@ ZEPHYRLIBS = $(KRB4_LDFLAGS) $(KRB4_LIBS
 
 ZEPHYRLIBSEXT = $(ZEPHYR_LDFLAGS) $(ZEPHYR_LIBS)
 
-libzephyr_la_LDFLAGS = -module -avoid-version
+libzephyr_la_LDFLAGS = -module -avoid-version $(PURPLE_PLUGIN_LDFLAGS)
 
 if STATIC_ZEPHYR
 
--- origsrc/pidgin-2.6.4/pidgin/Makefile.am	2009-11-29 21:13:54.000000000 -0600
+++ src/pidgin-2.6.4/pidgin/Makefile.am	2009-12-19 22:14:17.094904600 -0600
@@ -68,7 +68,7 @@ if ENABLE_GTK
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = pidgin.pc
 
-SUBDIRS = pixmaps plugins
+SUBDIRS = . pixmaps plugins
 
 bin_PROGRAMS = pidgin
 
@@ -194,7 +194,13 @@ pidgininclude_HEADERS = \
 
 
 pidgin_DEPENDENCIES = @LIBOBJS@
+
+if PLATFORM_WIN32
+pidgin_LDFLAGS = -Wl,--export-all-symbols,--out-implib,libpidgin.a
+else
 pidgin_LDFLAGS = -export-dynamic
+endif
+
 pidgin_LDADD = \
 	@LIBOBJS@ \
 	$(GLIB_LIBS) \
@@ -232,3 +238,9 @@ AM_CPPFLAGS = \
 	$(LIBXML_CFLAGS) \
 	$(INTGG_CFLAGS)
 endif  # ENABLE_GTK
+
+if PLATFORM_WIN32
+install-data-local:
+	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) libpidgin.a $(DESTDIR)$(libdir)
+endif
--- origsrc/pidgin-2.6.4/pidgin/pidgin.pc.in	2009-11-29 21:13:55.000000000 -0600
+++ src/pidgin-2.6.4/pidgin/pidgin.pc.in	2009-12-19 22:14:17.103905100 -0600
@@ -11,4 +11,5 @@ Description: Pidgin is a GTK2-based inst
 Version: @VERSION@
 Requires: gtk+-2.0 purple
 Cflags: -I${includedir}/pidgin
+Libs: -L${libdir} @LIBPIDGIN@
 
--- origsrc/pidgin-2.6.4/pidgin/plugins/Makefile.am	2009-11-29 21:13:57.000000000 -0600
+++ src/pidgin-2.6.4/pidgin/plugins/Makefile.am	2009-12-19 22:14:17.108905400 -0600
@@ -31,24 +31,24 @@ SUBDIRS = \
 
 plugindir = $(libdir)/pidgin
 
-convcolors_la_LDFLAGS       = -module -avoid-version
-contact_priority_la_LDFLAGS = -module -avoid-version
-extplacement_la_LDFLAGS     = -module -avoid-version
-gtk_signals_test_la_LDFLAGS = -module -avoid-version
-gtkbuddynote_la_LDFLAGS     = -module -avoid-version
-history_la_LDFLAGS          = -module -avoid-version
-iconaway_la_LDFLAGS         = -module -avoid-version
-markerline_la_LDFLAGS       = -module -avoid-version
-notify_la_LDFLAGS           = -module -avoid-version
-pidginrc_la_LDFLAGS         = -module -avoid-version
-relnot_la_LDFLAGS           = -module -avoid-version
-sendbutton_la_LDFLAGS       = -module -avoid-version
-spellchk_la_LDFLAGS         = -module -avoid-version
-themeedit_la_LDFLAGS        = -module -avoid-version
-timestamp_la_LDFLAGS        = -module -avoid-version
-timestamp_format_la_LDFLAGS = -module -avoid-version
-vvconfig_la_LDFLAGS         = -module -avoid-version
-xmppconsole_la_LDFLAGS      = -module -avoid-version
+convcolors_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+contact_priority_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+extplacement_la_LDFLAGS     = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+gtk_signals_test_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+gtkbuddynote_la_LDFLAGS     = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+history_la_LDFLAGS          = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+iconaway_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+markerline_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+notify_la_LDFLAGS           = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+pidginrc_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+relnot_la_LDFLAGS           = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+sendbutton_la_LDFLAGS       = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+spellchk_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+themeedit_la_LDFLAGS        = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+timestamp_la_LDFLAGS        = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+timestamp_format_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+vvconfig_la_LDFLAGS         = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
+xmppconsole_la_LDFLAGS      = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.6.4/pidgin/plugins/cap/Makefile.am	2009-11-29 21:13:57.000000000 -0600
+++ src/pidgin-2.6.4/pidgin/plugins/cap/Makefile.am	2009-12-19 22:14:17.112905600 -0600
@@ -1,6 +1,6 @@
 capdir = $(libdir)/pidgin
 
-cap_la_LDFLAGS = -module -avoid-version
+cap_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.6.4/pidgin/plugins/disco/Makefile.am	2009-11-29 21:13:58.000000000 -0600
+++ src/pidgin-2.6.4/pidgin/plugins/disco/Makefile.am	2009-12-19 22:14:17.117905900 -0600
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/pidgin
 
-xmppdisco_la_LDFLAGS = -module -avoid-version
+xmppdisco_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 EXTRA_DIST = \
         Makefile.mingw
--- origsrc/pidgin-2.6.4/pidgin/plugins/gestures/Makefile.am	2009-11-29 21:13:58.000000000 -0600
+++ src/pidgin-2.6.4/pidgin/plugins/gestures/Makefile.am	2009-12-19 22:14:17.121906100 -0600
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/pidgin
 
-gestures_la_LDFLAGS = -module -avoid-version
+gestures_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.6.4/pidgin/plugins/gevolution/Makefile.am	2009-11-29 21:13:58.000000000 -0600
+++ src/pidgin-2.6.4/pidgin/plugins/gevolution/Makefile.am	2009-12-19 22:14:17.125906300 -0600
@@ -1,6 +1,6 @@
 plugindir = $(libdir)/pidgin
 
-gevolution_la_LDFLAGS = -module -avoid-version
+gevolution_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
--- origsrc/pidgin-2.6.4/pidgin/plugins/musicmessaging/Makefile.am	2009-11-29 21:13:58.000000000 -0600
+++ src/pidgin-2.6.4/pidgin/plugins/musicmessaging/Makefile.am	2009-12-19 22:14:17.130906600 -0600
@@ -3,7 +3,7 @@ EXTRA_DIST = \
 
 musicmessagingdir = $(libdir)/pidgin
 
-musicmessaging_la_LDFLAGS = -module -avoid-version
+musicmessaging_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 if ENABLE_DBUS
--- origsrc/pidgin-2.6.4/pidgin/plugins/perl/common/Makefile.PL.in	2009-11-29 21:13:58.000000000 -0600
+++ src/pidgin-2.6.4/pidgin/plugins/perl/common/Makefile.PL.in	2009-12-19 22:14:17.135906900 -0600
@@ -10,6 +10,7 @@ WriteMakefile(
        'AUTHOR'        => 'Pidgin <http://pidgin.im/>') :  ()),
     'DEFINE'        => '@DEBUG_CFLAGS@',
     'INC'           => '-I. -I@srcdir@ -I@top_srcdir@ -I@top_srcdir@/libpurple -I@top_srcdir@/pidgin @GTK_CFLAGS@',
+    'LIBS'          => ['-L@top_builddir@/libpurple/.libs -lpurple -L@top_builddir@/pidgin @LIBPIDGIN@ @GLIB_LIBS@'],
     'OBJECT'        => '$(O_FILES)', # link all the C files too
     'TYPEMAPS'      => ["@top_srcdir@/libpurple/plugins/perl/common/typemap"],
 #    'OPTIMIZE'      => '-g', # For debugging.
--- origsrc/pidgin-2.6.4/pidgin/plugins/ticker/Makefile.am	2009-11-29 21:13:58.000000000 -0600
+++ src/pidgin-2.6.4/pidgin/plugins/ticker/Makefile.am	2009-12-19 22:14:17.139907100 -0600
@@ -3,7 +3,7 @@ EXTRA_DIST = \
 
 plugindir = $(libdir)/pidgin
 
-ticker_la_LDFLAGS = -module -avoid-version
+ticker_la_LDFLAGS = -module -avoid-version $(PIDGIN_PLUGIN_LDFLAGS)
 
 if PLUGINS
 
