pkg-config thinks 1.9.0-rc2 is newer than 1.9.0

--- origsrc/pidgin-2.7.5/configure.ac	2010-11-02 09:44:41.316029700 -0500
+++ src/pidgin-2.7.5/configure.ac	2010-11-02 10:21:57.201407800 -0500
@@ -1046,7 +1046,7 @@ else
 	gadu_manual_check="no"
 fi
 if test "x$gadu_manual_check" = "xno"; then
-	PKG_CHECK_MODULES(GADU, [libgadu >= 1.9.0-rc2], [
+	PKG_CHECK_MODULES(GADU, [libgadu >= 1.9.0], [
 		gadu_includes="yes"
 		gadu_libs="yes"
 	], [
@@ -1080,7 +1080,7 @@ if test "x$gadu_libs" = "xyes"; then
 	]])], [
 		AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <libgadu.h>]], [[
 #if GG_DEFAULT_PROTOCOL_VERSION < 0x2e
-#error "Your libgadu version is too old. libpurple requires 1.9.0-rc2 or higher."
+#error "Your libgadu version is too old. libpurple requires 1.9.0 or higher."
 #endif
 		]])], [
 			AC_MSG_RESULT(yes)
@@ -1091,7 +1091,7 @@ if test "x$gadu_libs" = "xyes"; then
 			echo
 			echo
 			echo "Your supplied copy of libgadu is too old."
-			echo "Install version 1.9.0-rc2 or newer."
+			echo "Install version 1.9.0 or newer."
 			echo "Then rerun this ./configure"
 			echo
 			echo "Falling back to using our own copy of libgadu"
