--- origsrc/pidgin-2.7.5/libpurple/plugins/ssl/ssl-nss.c	2010-06-16 23:00:33.000000000 -0500
+++ src/pidgin-2.7.5/libpurple/plugins/ssl/ssl-nss.c	2010-11-02 12:43:18.214480900 -0500
@@ -130,7 +130,9 @@ ssl_nss_init_nss(void)
 	NSS_NoDB_Init(".");
 
 	/* TODO: Fix this so autoconf does the work trying to find this lib. */
-#ifndef _WIN32
+#ifdef __CYGWIN__
+	lib = g_strdup("cygnssckbi.dll");
+#elif !defined _WIN32
 	lib = g_strdup(LIBDIR "/libnssckbi.so");
 #else
 	lib = g_strdup("nssckbi.dll");
